---
title: "Metabolic genesets"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(msigdbr)
genesets = readRDS("../data/hallmarks_genesets.rds")
library(tidyverse)
library(fgsea)
library(clusterProfiler)
```

```{r}
# Obtaining gene sets from msigdb for Carbohydrate metabolism
Homo_sapiensCPR = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CP:REACTOME")
Carbohydrates = Homo_sapiensCPR[Homo_sapiensCPR$gs_id=="M16864",]

# Glycolysis
Homo_sapiensCGP = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CGP")
Glycolysis = Homo_sapiensCGP[Homo_sapiensCGP$gs_id=="M16111",] #22 genes

# Amino acid 
Homo_sapiensC2 = msigdbr(species = "Homo sapiens", category = "C2")
Amino_Acid = Homo_sapiensC2[Homo_sapiensC2$gs_id=="M39570",] #102 genes

#Sphingolipid
Sphingolipid = Homo_sapiensCPR[Homo_sapiensCPR$gs_id=="M14857",]

#Fatty acid
Homo_sapiensCPK = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CP:KEGG")
Fatty_acid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M699",]

#Pentose phosphate
Pentose_phosphate = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M1386",]

#Ether lipid
Ether_lipid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M2130",]

#Glutathione
Glutathione = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M1840",]

#Nitrogen
Nitrogen = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M4629",]

#Oxidative phosphorylation
Oxidative_phosphorylation = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M19540",]

#Purines
Purine = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M14314",]

#Pyrimidines
Pyrimidine = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M5109",]

#Pyruvate
Pyruvate = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M7934",]

#Steroids
Steroid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M5872",]

#Steroid hormones
Steroid_hormones = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M14933",]


```

```{r}
#create ascending ranked gene vector for first BLCA patient
#extract column first patient BLCA
first = BLCA_z[,1, drop = FALSE]
#sort ascending according to z score, KEEPING gene names
first_sorted = first[order(first),, drop = FALSE]
#convert into ranked gene vector
first_BLCA = as.vector(first_sorted$`TCGA-MV-A51V-01`)
#retrieve names from before
first_ensemble_id = rownames(first_sorted)
names(first_BLCA) = first_ensemble_id
#turn gene names into NCBI entrez gene ID
mart = useEnsembl(dataset = "hsapiens_gene_ensembl", biomart='ensembl')
first_geneids = getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","entrezgene_id"),values= first_ensemble_id, mart= mart) #gives longer output vector than input???
sum(duplicated(first_geneids$ensembl_gene_id)) #64 double ensemble! (9847-9783 = 64)
#remove duplicates, then like original 9783 rows/genes
first_geneids = first_geneids[!duplicated(first_geneids$ensembl_gene_id),]
first_geneids = na.omit(first_geneids)# remove NA's --> now 8635 genes left

first_entrez_id = first_geneids$entrezgene_id #8635 entrez ids


#need to reorder entrez gene id's because original order is lost after biomart!
#ALSO: HOW REMOVE GENES WHERE NO ENTREZ ID WAS FOUND? THEY'RE NOT LEFT IN ENTREZ VECTOR!
#HOW MATCH POSITION OF GENE NAMES NOW TO RANKED GENE VECTOR??
#give vector elements according entrez gene ID as name

```

```{r}
#create GSEA compatible gene set list, NCBI entrez gene id as characters
carbohydrates = as.character(Carbohydrates$ensembl_gene)
glycolysis = as.character(Glycolysis$ensembl_gene)
amino_acid = as.character(Amino_Acid$ensembl_gene)
sphingolipid = as.character(Sphingolipid$ensembl_gene)
fatty_acid = as.character(Fatty_acid$ensembl_gene)
pentose_phosphate = as.character(Pentose_phosphate$ensembl_gene)
ether_lipid = as.character(Ether_lipid$ensembl_gene)
glutathione = as.character(Glutathione$ensembl_gene)
nitrogen = as.character(Nitrogen$ensembl_gene)
oxidative_phosphorylation = as.character(Oxidative_phosphorylation$ensembl_gene)
purine = as.character(Purine$ensembl_gene)
pyrimidine = as.character(Pyrimidine$ensembl_gene)
pyruvate = as.character(Pyruvate$ensembl_gene)
steroid = as.character(Steroid$ensembl_gene)
steroid_hormones = as.character(Steroid_hormones$ensembl_gene)

met.pathways = list(carbohydrates, glycolysis, amino_acid, sphingolipid, fatty_acid, pentose_phosphate, ether_lipid, glutathione, nitrogen, oxidative_phosphorylation, purine, pyrimidine, pyruvate, steroid, steroid_hormones)

names(met.pathways) = c("M16864_Carbohydrates", "M16111_Glycolysis", "M39570_Amino_acid", "M14857_Sphingolipid", "M699_Fatty_Acid", "M1386_Pentose_phosphate", "M2130_Ether_lipid", "M1840_Glutathione", "M4629_Nitrogen", "M19540_Oxidative_phosphorylation", "M14314_Purine", "M5109_Pyrimidine", "M7934_Pyruvate", "M5872_Steroid", "M14933_Steroid_hormones")
```



```{r}
#next adapt format to hallmark genesets and extract the genes from our filtered matrix
carbohydrates_genesymbols = Carbohydrates$human_gene_symbol
glycolysis_genesymbols = Glycolysis$human_gene_symbol
amino_Acid_genesymbols = Amino_Acid$human_gene_symbol
nucleotide_genesymbols = Nucleotide$human_gene_symbol

metabolic_pathways = list(
  carbohydrate = carbohydrates_genesymbols,
  glycolysis = glycolysis_genesymbols,
  amini_acid = amino_Acid_genesymbols,
  nucleotide = nucleotide_genesymbols)

```


```{r}
KIRC_tumor_cl = tcga_KIRC_tumor_protein_coding[rownames(tcga_KIRC_tumor_protein_coding) %in% rownames(tcga_KIRC_cl_pv_FC_sig),]

KIRC_norm_cl = tcga_KIRC_norm_protein_coding[rownames(tcga_KIRC_norm_protein_coding) %in% rownames(tcga_KIRC_cl_pv_FC_sig),]

tcga_KIRC_cl_pv_FC


```


```{r}
ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")
carb_ensembl = getBM(attributes='hgnc_symbol', 
      filters = 'ensembl_gene_id', 
      values = carbohydrates_genesymbols, 
      mart = ensembl)
dim(carb_ensembl)

```


```{r}
#Enrichment analysis using fisher exact test 
carbohydrate_enricher = select(Carbohydrates, gs_name, ensembl_gene)
enrich.c = enricher(gene = rownames(KIRC_tumor_cl), TERM2GENE = carbohydrate_enricher )
head(enrich.c@result)
#Gene ratio : number of sig.genes in the gene set 
```

```{r}
#actual gsea
#create a df with ensembl ids and log FC
tcga_KIRC_GSEA_prep = 
rownames_to_column(tcga_KIRC_cl_pv_FC, var = "ensembl_id")
tcga_KIRC_GSEA_prep = tcga_KIRC_GSEA_prep[,1:2]

tcga_KIRC_tumor = data.matrix(tcga_tumor_norm[["KIRC"]][["tumor"]])
tcga_KIRC_norm = data.matrix(tcga_tumor_norm[["KIRC"]][["normal"]])

FC.vec = tcga_KIRC_GSEA_prep[,2]
names(FC.vec) = tcga_KIRC_GSEA_prep$ensembl_id

tcga_KIRC_GSEA = fgseaSimple(pathways = met.pathways,
                             stats = FC.vec,
                             scoreType = "std",
                             nperm = 10000)

```
```{r}
tcga_KIRC_GSEA %>%
  filter(padj <= 0.05) %>%
  mutate(met.pathways = gsub ("metabolic pathways","",pathway),
  pathway = gsub("_"," ",pathway)) %>%
  
  ggplot(aes(x=reorder(pathway,NES), y=NES)) +
  geom_col ()+
  theme_classic()+
  lims(y=c(-3.2,3.2))+
  coord_flip()+
  labs (y = "Normalized enrichment score (NES)",
        x = "Gene set",
        title = "GSEA")

tcga_KIRC_GSEA1 %>%
  filter(padj <= 0.05) %>%
  mutate(met.pathways = gsub ("metabolic pathways","",pathway),
  pathway = gsub("_"," ",pathway)) %>%
  
  ggplot(aes(x=reorder(pathway,NES), y=NES)) +
  geom_col ()+
  theme_classic()+
  lims(y=c(-3.2,3.2))+
  coord_flip()+
  labs (y = "Normalized enrichment score (NES)",
        x = "Gene set",
        title = "GSEA")

plotEnrichment(met.pathways[["M16864_Carbohydrates"]],fc.order)

```


```{r}
#log2FC for every patient, only protein coding without constant genes
KIRC_FC_patients = data.frame(tcga_KIRC_tumor_cl - tcga_KIRC_norm_cl)
KIRC_GSEA_prep = 
rownames_to_column(KIRC_FC_patients, var = "ensembl_id")

FC.vec = tcga_KIRC_GSEA_prep[,2]
names(FC.vec) = tcga_KIRC_GSEA_prep$ensembl_id
fc.order = FC.vec[order(FC.vec),drop = FALSE]

tcga_KIRC_GSEA = fgsea(pathways = met.pathways,
                             stats = fc.order,
                             scoreType = "std")
tcga_KIRC_GSEA1 = fgsea(pathways = met.pathways,
                             stats = FC.vec,
                             scoreType = "std")
y = (for (i in 2:(ncol(KIRC_GSEA_prep))){
  data.frame(fgseaSimple(pathways = met.pathways,
                             stats =  (names(KIRC_GSEA_prep[,i]) =  KIRC_GSEA_prep$ensembl_id),
                             scoreType = "std",
                             nperm = 10000))
})
  
KIRC_GSEA_results = list()
(for (i in 2:(ncol(KIRC_GSEA_prep))){
  x = c(KIRC_GSEA_prep[,i])
  names(x) = KIRC_GSEA_prep$ensembl_id
  x = x[order(x),drop = FALSE]
  KIRC_GSEA_results[[i]] = fgsea(pathways = met.pathways,
                             stats =  x,
                             scoreType = "std")
})


```


