---
title: " "
output:
  pdf_document:
    number_sections: TRUE
  html_document:
    code_folding: hide
documentclass: scrreprt
classoption: [parskip, oneside] # für drucken twoside
header-includes:
  - \usepackage[ ngerman, main=english]{babel}
  - \usepackage[utf8]{inputenc}
  - \usepackage[T1]{fontenc}
  - \usepackage{lmodern}
  - \usepackage[onehalfspacing]{setspace}
  - \usepackage[left=2.50cm, right=2.50cm, top=2.50cm, bottom=2.50cm, bindingoffset=10mm, includehead, includefoot]{geometry}
  - \usepackage[headsepline]{scrlayer-scrpage}
  - \usepackage{url}
  - \usepackage[backend=biber, style=authoryear, giveninits=true, maxbibnames=99, uniquename=init, maxcitenames=2, hyperref=true, date=year]{biblatex}
  - \usepackage{xpatch}
  - \usepackage{csquotes}
  - \usepackage{amsmath}
  - \usepackage{listings}
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{multirow}
  - \usepackage{rotating}
  - \usepackage{subfigure}
  - \usepackage{graphicx}
  - \usepackage{float}
  - \usepackage{acronym}
  - \usepackage{lipsum}
  - \usepackage{scrhack}
  - \emergencystretch=50pt
  - \clubpenalty = 10000
  - \widowpenalty = 10000
  - \displaywidowpenalty = 10000
  - \automark[section]{chapter}
  - \renewcommand*{\chaptermarkformat}{}
  - \renewcommand*{\sectionmarkformat}{}
  - \setkomafont{title}{\sffamily}
  - \setkomafont{disposition}{\usekomafont{title}}
  - \setkomafont{author}{\usekomafont{title}}
  - \setkomafont{date}{\usekomafont{title}}
  - \setkomafont{caption}{\sffamily\small}
  - \setkomafont{captionlabel}{\usekomafont{caption}\bfseries\small}
  - \setkomafont{pagehead}{\normalfont\scshape}
csl: bib/molecular-biology-of-the-cell.csl
bibliography: [bib/packages.bib, bib/BibTeX.bib]
---
<!-- Based on Nicolas Peschkes LaTeX Template for MoBis, if you want further information on the packages just check it here: https://gitlab.com/npeschke/LaTeX-Template-IPMB -->

<!-- your title page -->

\begin{titlepage}
\centering
	{\Large Ruprecht-Karls-Universität Heidelberg\\
		Fakultät für Biowissenschaften\\
		Bachelorstudiengang Molekulare Biotechnologie\\}

	{\vspace{\stretch{2}}}
	{\usekomafont{title}

		{\Huge ksdflsdjf}

		{\Huge sfdsfd}

		{\Huge sfsf}

	}

	\vspace{\stretch{2}}
	{\Large Data Science Project SoSe 2022}

	\vspace{\stretch{2}}

	{\Large
		\begin{tabular}{rl}
			Autoreb & Max Mustermann, jklsdfksldkfslkdjf\\
			Geburtsort & sdfjksafdls\\
			Abgabetermin &20.07.2022\\
		\end{tabular}
	}

	\vspace{\stretch{1}}

\end{titlepage}


<!-- add your chapters here -->
\tableofcontents

<!-- Intoduction -->
# Introduction


## Computational Tools

### Dimension reduction

#### PCA

#### UMAP 


### Statistical analysis

#### Shapiro-Wilks test
Shapiro-Wilks (SW) test is a normality test based on regression and correlation. It tests the null hypothesis that the data follows a normal distribution. Small values of SW test statistic indicate no normality of the data thus the null hypothesis is rejected. SW values of one suggest normality [@Shapiro-Yap-2011].

#### Wilcoxon rank-sum and signed-rank test
Wilcoxon rank-sum test and Wilcoxon signed-rank test both are non-parametric statistical hypothesis tests that can be used when the data does not follow a normal distribution.  Wilcoxon signed-rank test is used to analyze matched-pair or one-sample data. It tests the null hypothesis that there is no difference in probability distribution of first and second sample, hence the distribution of pairwise differences is centered at zero. The test is based on ranked absolute values of differences @Wilcoxon-Woolson-2007. Wilcoxon rank-sum test is performed when analyzing unpaired-data and is likewise based on ranked values. The null hypothesis states that there is no association between the two samples @Wilcoxon-Rey2011. 

#### Kruksal-Wallis 
Kruksal-Wallis test is a rank-based non-parametric hypothesis test. It is an extension of Wilcoxon rank-sum test and can allows comparing more than two independent data sets. Kruksal-Wallis test tests the null hypothesis that there is no difference in distributions of all $k$ data sets. Alternative hypothesis states that there is a difference in at least two populations **(oder: …that at least two populations show stochastic heterogeneity)** @Kruksal-Wallis-ostertagova-2014. 

#### Bonferroni correction
Multiple statistical testing results in an increased risk for type I errors (false positives). Bonferroni correction is used to reduce this type I error rate. For this, the significance level is adjusted by dividing the critical $p$-value $\alpha$ by the number of tests @Bonferroni-Amstrong-2014. 

### Clustering

#### Kmeans

#### Hierarchial clustering

### Immune deconvolution

The immune deconvolution package (@immunedeconv-merotto-2022) is used to obtain immune cell fractions from bulk RNA-sequencing data. The input is a matrix with genes as rows and samples as columns, containing gene expression values as transcripts per million (TPM). The deconvolution algorithm models the expression of a single gene as a linear combination of the expression of that gene across the different cell types. An equation for each gene is set up that contains terms of the matrix multiplication of a signature matrix $S$ and an immune cell fraction vector $F$. The signature matrix $S$ contains all average gene expression profiles for each gene in the immune cell types, respectively. The output using the “quanTIseq” method is a matrix containing the immune cell fractions for each sample that have been re-normalized to sum up to one (@Immune-deconvolution-Finotello-2018).

### GSEA


### Binary logistic regression

Binary logistic regression is a statistical method used to predict a dichotomous dependent variable from one or several independent variables. In contrast to linear regression, logistic regression does not require a linear relationship between the independent and the dependent variable. 

A logistic regression model is created by training on one dataset and testing it on another independent one. First, the observations of the dependent variable of the training dataset are converted into $0$ an $1$ for each outcome, respectively. These data points can be plotted into a coordinate system and subsequently be moved to positive and negative infinity through a $log(odds)$-transformation. The odds ratio is calculated using the logit function:
\begin{gather*}
log(odds)=log(\frac{p}{1-p})
\end{gather*}
A regression line is now fitted to these data points but can’t be optimized by least square method, as in linear regression, since the residuals strive towards infinity. Alternatively, the data points are projected onto the fitted line and thus get a new $log(odds)$-value. In order to plot the sigmoidal logistic graph, the new $log(odds)$ value of each data point is inserted into the logistic function, which can be derived by rearranging the $log(odds)$-equation:
\begin{gather*}
p=\frac{e^{log(odds)}}{1+e^{log(odds)}}
\end{gather*}
The result of the function is the probability between $0$ and $1$ for each data point to have a $1$ as the outcome of the dependent variable. The best fitting linear regression line can be found using the “maximum likelihood”, which is either computed by the product of the probability of all data points to have the observation $1$ or the sum of the logarithmic probabilities. The calculation of the likelihood is made for all possible regression lines, by rotating the line minimally for every new computation. The best fitting line is the one with the maximum value of the likelihood. 

For the evaluation of the regression model $R^2$-values and statistical tests can be considered. Since the residuals cannot be used to compute $R^2$ in logistic regression, several pseudo-$R^2$, such as the “McFadden’s” were introduced, which compare the maximum likelihood of the model to a null model. A Wald-test and a $chi2$-test can be performed to test for the significance of each coefficient and of the overall model (@LogReg-Sperandei-2014, @LogReg-Peng-2002).


<!-- Materials and Methods -->

# Materials and Methods

## Data cleaning

The analysis was focused on two data sets containing **bulk-cell**(?) sequencing data. Pan-cancer analysis was performed on gene expression data of 9783 patients of 33 tumor types (Pan-cancer data set (DS1)). Focused analysis was based on gene expression data of normal and tumor tissue of 72 KIRC patients (DS2). The data obtained was already normalized by $log2$(Transcripts per million).
Both data sets were filtered for protein-coding genes using biomaRt package. This reduced gene expression data in DS1 from 60 498 to 19 624 genes and in DS2 from 19624 to 19186 genes. In DS1, variance was computed for each gene over all samples and the lower p50-quantile was subsequently removed. Furthermore, constantly expressed genes only were removed in DS2, resulting in gene expression data of 18 645 genes.

## TCGA pan-cancer analysis

## Focused Analysis: Identifying subtypes in KIRC
For focused analysis, KIRC was examined in more detail. The analysis was based on DS2. Data cleaning was performed as described above.  
First, differential gene expression in tumor tissue compared to normal tissue was analyzed by calculation of Foldchange $FC =(mean condition 1) / (mean condition 2)$ for each gene, where condition one represents gene expression of tumor tissue and condition two gene expression of normal tissue. Statistical significance was determined using Wilcoxon signed-rank test with Bonferroni correction, as Shapiro-Wilks test indicated no normality of the data.  
The second part of this analysis was focused on differential pathway activity. Pathway activity matrices were determined using GSEA, where genes were ranked based on their foldchange. Pathways with an adjusted $p$-value $>0.05$ were considered as non-significant, hence their NES was set to zero. Hallmark-pathways, KEGG-pathways and PID-pathways were analyzed. Patients were compared based on their pathway activity. Subclusters of patients were visualized for each geneset by UMAP calculated on PCA-results and subsequently identified by k-means clustering. Optimal number of clusters was determined using elbow method and silhouette method. Pathways that were crucial for this clustering and were significantly different between those clusters were detected by Wilcoxon rank-sum test for two clusters and Kruksal-Wallis test for three clusters, both with Bonferroni correction.

## Predicting immune infiltration with logistic regression
A binary logistic regression model was created based on the pathway activity of KIRC samples from DS1 and tested on DS2. The dependent variable to be predicted was the “Immune infiltration” of samples. The independent variables were pathways chosen from the most differentially significantly expressed PID pathways between the three clusters that have emerged in the PID clustering of KIRC patients in DS1. All patients of the cluster with highest pathway activity were marked as immune infiltrated with a “$1$” and all other patients with a “$0$”. The logistic regression was performed to predict the immune infiltration in the samples of DS2.

## Packages

<!-- Results -->

# Results

## TCGA pan-cancer analysis

## KIRC specific analysis

## Logistic regression 

\begin{tabular}{ |p{4.5cm}||p{3cm}|p{3cm}|p{3cm}|  }
 \hline
 \multicolumn{4}{|c|}{Logistic regression model results} \\
 \hline
  & \textbf{Estimate} & \textbf{Standard error} & \textbf{p-value}\\
 \hline
 Intercept   &  $-2.5856$  & $0.2059$ &   $<2e-16$\\
 PID\_TCR\_PATHWAY &   $0.9727$  & $0.2027$   & $1.59e-06$\\
 PID\_IL1\_PATHWAY & $1.1651$ & $0.1934$ &  $1.70e-09$\\
\hline
\end{tabular}


<!-- Discussion -->
# Discussion

<!-- Concluding remarks/Outlook -->

# Concluding remarks/Outlook




# References

<div id="refs"></div>

# Appendix
