---
title: " "
output:
  pdf_document:
    number_sections: TRUE
  html_document:
    code_folding: hide
documentclass: scrreprt
classoption: [parskip, oneside] # für drucken twoside
header-includes:
  - \usepackage[ ngerman, main=english]{babel}
  - \usepackage[utf8]{inputenc}
  - \usepackage[T1]{fontenc}
  - \usepackage{lmodern}
  - \usepackage[onehalfspacing]{setspace}
  - \usepackage[left=1.50cm,right=1.50cm,top=1.50cm,bottom=1.50cm,includehead,includefoot]{geometry}
  - \usepackage[headsepline]{scrlayer-scrpage}
  - \usepackage{url}
  - \usepackage[backend=biber, style=authoryear, giveninits=true, maxbibnames=99, uniquename=init, maxcitenames=2, hyperref=true, date=year]{biblatex}
  - \usepackage{xpatch}
  - \usepackage{cleveref}
  - \usepackage{csquotes}
  - \usepackage{amsmath}
  - \usepackage{listings}
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{multirow}
  - \usepackage{rotating}
  - \usepackage{subfigure}
  - \usepackage{graphicx}
  - \usepackage{float}
  - \usepackage{acronym}
  - \usepackage{lipsum}
  - \usepackage{scrhack}
  - \emergencystretch=50pt
  - \clubpenalty = 10000
  - \widowpenalty = 10000
  - \displaywidowpenalty = 10000
  - \automark[section]{chapter}
  - \renewcommand*{\chaptermarkformat}{}
  - \renewcommand*{\sectionmarkformat}{}
  - \setkomafont{title}{\sffamily}
  - \setkomafont{disposition}{\usekomafont{title}}
  - \setkomafont{author}{\usekomafont{title}}
  - \setkomafont{date}{\usekomafont{title}}
  - \setkomafont{caption}{\sffamily\small}
  - \setkomafont{captionlabel}{\usekomafont{caption}\bfseries\small}
  - \setkomafont{pagehead}{\normalfont\scshape}
csl: bib/molecular-biology-of-the-cell.csl
bibliography: [bib/packages.bib, bib/BibTeX.bib]
editor_options: 
  markdown: 
    wrap: sentence
---

<!-- Based on Nicolas Peschkes LaTeX Template for MoBis, if you want further information on the packages just check it here: https://gitlab.com/npeschke/LaTeX-Template-IPMB -->

<!-- your title page -->

```{=tex}
\begin{titlepage}
\centering
    {\Large Ruprecht-Karls-Universität Heidelberg\\
        Fakultät für Biowissenschaften\\
        Bachelorstudiengang Molekulare Biotechnologie\\}

    {\vspace{\stretch{2}}}
    {\usekomafont{title}

        {\huge Molecular Characterization of Immune Signaling and Metabolic Subtypes in Human Cancer}

        {\Huge }
        
        {\LARGE A Pan-Cancer Analysis with therapeutic outlooks for clear cell renal cell carcinoma}

    }

    \vspace{\stretch{2}}
    {\Large Data Science Project SoSe 2022}

    \vspace{\stretch{2}}

    {\Large
        \begin{tabular}{rl}
            Autoren & Anna von Bachmann, Linda Blaier, Maja Glotz, Tim Wenzel\\
            Heidelberg &18.07.2022\\
        \end{tabular}
    }

    \vspace{\stretch{1}}

\end{titlepage}
```
\addchap{Abstract}

\tableofcontents

\addchap{List of Abbreviations}

\small

```{=tex}
\begin{tabular}{rl}
        ACC & Adrenocortical carcinoma\\
        BLCA & Bladder Urothelial Carcinoma\\
        BRCA & Breast invasive carcinoma\\
        CESC & Cervical squamous cell carcinoma and endocervical adenocarcinoma\\
        CHOL & Cholangiocarcinoma\\
        COAD & Colon adenocarcinoma\\
      DLBC & Lymphoid Neoplasm Diffuse Large B-cell Lymphoma\\
      ESCA & Esophageal carcinoma\\
      FDR & False discovery rate\\ 
      GBM & Glioblastoma multiforme\\
      GSEA & Gene set enrichment analysis\\
      HNSC & Head and Neck squamous cell carcinoma\\
      KEGG & Kyoto encyclopedia of genes and genomes\\
      KICH & Kidney Chromophobe\\
      KIRC & Kidney renal clear cell carcinoma\\
      KIRP & Kidney renal papillary cell carcinoma\\
      LAML & Acute Myeloid Leukemia\\
      LGG    & Brain Lower Grade Glioma\\
      LIHC & Liver hepatocellular carcinoma\\
      LUAD & Lung adenocarcinoma\\
      LUSC & Lung squamous cell carcinoma\\
      MESO & Mesothelioma\\
      OV & Ovarian serous cystadenocarcinoma\\
      PAAD & Pancreatic adenocarcinoma\\
      PC & Principal component\\
      PCA & Principal component analysis\\
      PCPG & Pheochromocytoma and Paraganglioma\\
      PID & Pathway interaction database\\
      PRAD & Prostate adenocarcinoma\\
      RCC & Renal cell carcinoma\\
      READ & Rectum adenocarcinoma\\
      SARC & Sarcoma\\
      SKCM & Skin Cutaneous Melanoma\\
      STAD & Stomach adenocarcinoma\\
      SW & Shapiro-Wilks\\
      TCGA & The cancer genome atlas\\
      TGCT & Testicular Germ Cell Tumors\\
      THCA & Thyroid carcinoma\\
      THYM & Thymoma\\
      TME & Tumor Microenvironment\\
      UCEC & Uterine Corpus Endometrial Carcinoma\\
      UC & Uterine Carcinosarcoma\\
      UMAP & uniform manifold approximation and projection\\
      UVM & Uveal Melanoma\\
        \end{tabular}
```
# Introduction

Cancer is one of the most prevalent diseases in the world and responsible for almost 10 million deaths each year [@cancer_statistics_Ferlay_2021].
Therefore, it is a big interest to further understand the characteristics of cancer in more detail hoping to find new treatment options.
Pan cancer analysis aims to find similarities and differences across different tumor types.
These types of analysis are often based on The Cancer Genome Atlas (TCGA) which is a comprehensive genome sequencing database consisting of bulk samples of over 11 000 tumors from the 33 most prevalent forms of cancer[@TCGA_Cooper_2018].

It is known that cancer cells have certain characteristics that distinguish them from normal cells, the so-called hallmarks of cancer.
These enable them to evade human protection mechanisms and sustain uncontrolled cell proliferation.
Cancer hallmarks often are a consequence of altered gene expression patterns [@cancer_hallmarks_hanahan_2011].
Among other things in this project, we want to identify and compare pathways with altered gene expression who are primarily responsible for establishing the cancer hallmarks across the 33 tumor types selected in the TCGA.

The activation of oncogenes and the loss of tumor suppressor gene functions are the main cause for tumor development.
Both can lead to reprogramming of cellular metabolism.
This way cancer cells can cover their high needs for nutrients even in a nutrient-poor environment.
For example, cancer cells have an increased glucose metabolism due to the Warburg effect [@Metabolism_Warburg_1927; @metabolism_hallmarks_natalya_2016].
Therefore, we also did a pan cancer analysis on metabolic pathways for which we used pathways curated by the Kyoto Encyclopedia of Genes and Genomes (KEGG) [@KEGG_Ogata_1999].

Beyond the tumor cells themselves, a tumor type is also characterized by its microenvironment (TME).
Dependent on the composition of immune cells in the TME, it can either promote or suppress tumorigenesis.
To investigate immune signaling, we used a gene set from the Pathway interaction database (PID).
The respective pathways mainly concern signaling especially regarding cell cycle and immune response [@PID_Schaefer_2008].

Kidney cancer is one of the most common kinds of cancer.
Most kidney cancer patients are diagnosed with Renal Cell Carcinoma (RCC) of which renal clear cell carcinoma (KIRC) is the most prevalent form.
The treatment of choice for early stage RCC patients is surgical removal [@KIRC_Chen_2020].
While this is relatively successful with an overall survival of 60-70%, the prognosis for late stage RCC is less than 10%.
This can be attributed to RCC being an intratumorally heterogenous form of cancer.
Additionally, KIRC is resistant to traditional treatment methods like chemotherapy and radiotherapy [@cross-omics_KIRC_Dimitrieva_2016].
Therefore, novel individualized treatment options need to be developed [@Immune_infiltration_RCC_Zhang_2019].

## Star Methods

### Uniform Manifold Approximation and Projection

`UMAP` is a new developed dimensionality reduction technique which is very fast and can preserve global structure of data very well.
For that `UMAP` constructs a high-dimensional graph which is then used to create a low-dimensional graph to visualize the dataset.
The initial graph is created based on the calculation of Similarity Scores dependent on the number of nearest neighbors each data point should have.
From that, the low-dimensional representation is created and iteratively optimized so the structure of the initial dataset is preserved best which is also done by calculating Similarity Scores [@UMAP_McInnes_2018; @UMAP_Becht_2019].

### Wilcoxon rank-sum and signed-rank test

`Wilcoxon` rank-sum test and `Wilcoxon` signed-rank test both are non-parametric statistical hypothesis tests that can be used when the data does not follow a normal distribution.
`Wilcoxon` signed-rank test is used to analyze matched-pair or one-sample data.
It tests the null hypothesis that there is no difference in probability distribution of first and second sample, hence the distribution of pairwise differences is centered at zero.
The test is based on ranked absolute values of differences [@Wilcoxon-Woolson-2007].
`Wilcoxon` rank-sum test is performed when analyzing unpaired-data and is likewise based on ranked values.
The null hypothesis states that there is no association between the two samples [@Wilcoxon-Rey2011].

### Kruksal-Wallis

`Kruksal-Wallis` test is a rank-based non-parametric hypothesis test.
It is an extension of `Wilcoxon` rank-sum test and allows comparing more than two independent datasets.
`Kruksal-Wallis` test tests the null hypothesis that there is no difference in distributions of all $k$ data sets.
The alternative hypothesis states that at least two populations show stochastic heterogeneity [@Kruksal-Wallis-ostertagova-2014; @Kruksal-wallis-test-Vargha-1998].

### Bonferroni correction

Multiple statistical testing results in an increased risk for type I errors (false positives).
`Bonferroni` correction is used to reduce this type I error rate.
For this, the significance level is adjusted by dividing the `False decovery rate` (`FDR`) by the number of tests [@Bonferroni-Amstrong-2014].

### Immune deconvolution The immune deconvolution package [@immunedeconv-merotto-2022] is used to obtain immune cell fractions from bulk RNA-sequencing data.

The input is a matrix with genes as rows and samples as columns, containing gene expression values as transcripts per million (TPM).
The deconvolution algorithm models the expression of a single gene as a linear combination of the expression of that gene across the different cell types.
An equation for each gene is set up that contains terms of the matrix multiplication of a signature matrix $S$ and an immune cell fraction vector $F$.
The signature matrix $S$ contains all average gene expression profiles for each gene in the immune cell types, respectively.
The output using the `quanTIseq` method is a matrix containing the immune cell fractions for each sample that have been re-normalized to sum up to one [@Immune-deconvolution-Finotello-2018].

### Gene Set Enrichment Analysis

Gene set enrichment analysis (`GSEA`) is algorithm used for interpretation of gene expression data based on assembly of genes with similar biological functions, a so called gene set S. `GSEA` compares gene expression data between two different phenotypic  groups. Based on these phenotypic labels a ranked list of genes using a ranking metric is computed, with the goal of identifying gene sets that are not randomly distributed among the ranked gene list. For this purpose, 3 main steps are performed.
Firstly, the Enrichment Score (`ES`) is calculated using random walk: Whenever a gene of the gene set is present in the ranked gene list, the running-sum increases. Whenever, this is not the case, the running-sum decreases. Increase and decrease are weighted upon the rank of each gene. The `ES(S)` of the gene set is equal to the maximum difference from zero. Secondly, `Kolmogorov-Smirnov`test is used to compare the distribution of the genes in a gene set to the distribution of genes not included in the gene set. Additionally, the input data is permuted, creating a null distribution of ES. By comparing the observed `ES(S)` to the null distribution, the $p-value$ is determined. Observed `ES(S)` and null distribution are normalized by diving by each mean. For this purpose, positive and negatives scores are treated separately.
Lastly, an adjustment for multiple hypothesis testing is added. Therefore, `Benjamini-Hochberg` (`BH`) correction is used.  `BH` correction ranks the $p-values$ of the ES in increasing order and then calculated a correction factor by dividing the rank of the $p-value$ by the number of tests performed and the `false discovery rate` (`FDR`). Only is both correction factor and p-value are smaller than a self-imposed cutoff, the corresponding `ES` can be considered significant[@GSEA].

### Binary logistic regression

Binary logistic regression is a statistical method used to predict a dichotomous dependent variable from one or several independent predictors.
In contrast to linear regression, logistic regression does not require a linear relationship between the independent and the dependent variable.\
A logistic regression model is created by training on one dataset and testing it on another independent one.
First, the observations of the dependent variable of the training dataset are converted into $0$ an $1$ for each outcome, respectively.
These data points can be plotted into a coordinate system and subsequently be moved to positive and negative infinity through a $log(odds)$-transformation.
The odds ratio is calculated using the logit function $log(odds)=log(\frac{p}{1-p})$.\
A regression line is now fitted to these data points and these are projected onto the fitted line and thus get a new $log(odds)$-value.
In order to plot the sigmoidal logistic graph, the new $log(odds)$ value of each data point is inserted into the logistic function, which can be derived by rearranging the $log(odds)$-equation $p=\frac{e^{log(odds)}}{1+e^{log(odds)}}$.\
The result of the function is the probability between $0$ and $1$ for each data point to have a $1$ as the outcome of the dependent variable.
The best fitting linear regression line can be found using the "maximum likelihood" that is computed with the probabilities of each point.
The calculation of the likelihood is made for all possible regression lines, by rotating the line minimally for every new computation.
The best fitting line is the one with the maximum value of the likelihood.\
For the evaluation of the regression model $R^2$-values and statistical tests can be considered.
Since the residuals cannot be used to compute $R^2$ in logistic regression, several pseudo-$R^2$, such as the "McFadden's" were introduced, which compare the maximum likelihood of the model to a null model.
A Wald-test and a $chi2$-test can be performed to test for the significance of each coefficient and of the overall model [@LogReg-Sperandei-2014; @LogReg-Peng-2002].

<!-- Materials and Methods -->

# Materials and Methods

## Data cleaning

The analysis was focused on two data sets containing bulk-cell sequencing data.
Pan-cancer analysis was performed on gene expression data of 9783 patients of 33 tumor types (`DF1`).
Focused analysis was based on gene expression data of normal and tumor tissue of 72 KIRC patients (`DF2`).
The data obtained was already normalized by $log2$(TPM).\
Both data sets were filtered for protein-coding genes using `biomaRt` package.
This reduced gene expression data in `DF1` from 60 498 to 19 624 genes and in `DF2` from 19 624 to 19 186 genes.
In `DF1`, variance was computed for each gene over all samples and the lower p50-quantile was subsequently removed.
Furthermore, constantly expressed genes only were removed in `DF2`, resulting in gene expression data of 18 645 genes.

## TCGA pan-cancer analysis

After the data cleaning, as described above, the pan-cancer analysis was performed on `DF1`.
The dimension reduction methods `PCA` and `UMAP` were used to visualize patients from all tumor types in a two-dimensional space.
For further analysis, hallmarks, KEGG and PID gene sets were currated from the Molecular Signature Database[@MSigDB]
In order to determine the activity of these genesets in each patient, `GSEA` was carried out for each tumor type, respectively.
Genes were ranked according to the $z$-score after $z$-normalization of each gene across all samples within every tumor type according to [@PENG].
Pathways with an adjusted $FDR$ $>0.05$ using `Benjamini & Hochberg` correction method were considered as non-significant and their `NES` was set to zero.
Pathway activity matrices that contained these `NES` were generated for all gene sets and could be depicted in heatmaps.
Starting from these matrices, `hierarchical clustering` was performed to find groups of tumor types with similar regulation.
The variance of the`NES` for each pathway across all tumor types was computed in order to identify the top 30 pathways with the highest variance and hence most differences in regulation between tumors.
Results of the `GSEA` with PID pathways were analyzed in more detail by `hierarchical clustering` of patients based on the `NES` of PID pathways within each tumor type.
Significantly different enriched pathways between the resulting clusters were identified using `Kruksal-Wallis` test ($FDR$: $5\%$) and `Bonferroni` correction.\
For all KIRC patients of `DF1` the immune cell fractions of the bulk samples were estimated using the `ìmmunedeconv` package.
The `quanTIseq` method was used and therefore normalized gene expression values were transformed back into TPM.

## Focused Analysis: Identifying subtypes in KIRC

For focused analysis, KIRC was examined in more detail.
The analysis was based on `DF2`.
Data cleaning was performed as described above.\
First, differential gene expression in tumor tissue compared to normal tissue was analyzed by calculation of the Fold change $FC =(mean condition 1) / (mean condition 2)$ for each gene, where condition one represents gene expression of tumor tissue and condition two gene expression of normal tissue.
Statistical significance was determined using `Wilcoxon` signed-rank test with `Bonferroni` correction, as `Shapiro-Wilks` test indicated no normality of the data.\
The second part of this analysis was focused on differential pathway activity.
Pathway activity matrices were determined using `GSEA`, where genes were ranked based on their Fold change.
Pathways with an adjusted $p$-value $>0.05$ were considered as non-significant, hence their `NES` was set to zero.
Hallmark pathways, KEGG pathways and PID pathways were analyzed.
For PID and KEGG pathways, patients were clustered based on their pathway activity.
Subclusters of patients were visualized for each gene set by `UMAP` calculated on `PCA`-results and subsequently identified by `k-means` clustering.
Optimal number of clusters was determined using `elbow` method and `silhouette` method.
Pathways that were crucial for this clustering and were significantly different between those clusters were detected by `Wilcoxon` rank-sum test for two clusters and `Kruksal-Wallis` test for three clusters, both with `Bonferroni` correction.

## Predicting immune infiltration with logistic regression

A binary logistic regression model was created based on the pathway activity of KIRC samples from `DF1` and tested on `DF2`.
The dependent variable to be predicted was the "Immune infiltration" of samples.
The independent variables were pathways chosen from the most differentially significantly expressed PID pathways between the three clusters that have emerged in the PID clustering of KIRC patients in `DF1`.
The list of independent variables can be seen in Appendix **???** All patients of the cluster with highest pathway activity were marked as immune infiltrated with a "$1$" and all other patients with a "$0$".
The logistic regression was performed to predict the immune infiltration in the samples of `DF2`.

## Packages

The analysis was performed using Rversion 4.2.0 and Rstudio.
Packages that have been used are listed in Appendix **????** <!-- Results -->

# Results

## TCGA pan-cancer analysis

### GSEA deciphers differences in immune and metabolic signaling across tumor types


```{=tex}
\begin{figure}[h]
 \begin{center}
   \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel0_Ebene 1.pdf}
  \end{center}
  \caption{\textbf{Clustering of 33 tumor types based on gene expression} (A) `UMAP` besed on gene expression levels show clustering between different tumor types (B) Venn-diagram shows overlap between the significantly differentially expressed genes and genes in the gene sets Hallmark, KEGG and PID}
  \label{UMAP}
\end{figure}
```

Visualizing the 33 tumor types based on gene expression using `PCA` and `UMAP` revealed that gene expression landscape is highly diverse.
Nontheless, tumor types, which all differ severely in their phenotype and histopathology, appear to form clusters upon gene expression (see \Cref{UMAP} $A$).
To asses the foundation of these clusters, we utilized several gene sets, including hallmark, PID and KEGG, which include `24,7%` of our highly differentially expressed genes and were used for further analysis using `GSEA` (see \Cref{UMAP} $B$).

```{=tex}
\begin{figure}[h]
 \begin{center}
   \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel6_v2_Ebene_1.pdf}
  \end{center}
  \caption{\textbf{GSEAresults for 33 tumor types} Heatmap showing top 30 variant gene sets}
  \label{hm}
\end{figure}
```

`GSEA` was performed on `DF1` showing diverse pathway patterns across tumor types.
In order to visualize the enrichment results, heatmaps of the pathway activity matrices were plotted.
Four main clusters of the 33 tumor types could be identified in the hallmark `GSEA`.
In general, many pathways did not show a remarkable enrichment across all tumor types.
Strikingly, immune response associated pathways, like Interferon and Interleukin singaling, were significantly upregulated in PAAD, OV and STAD and downregulated in LGG, UVM and ACC.
Moreover, pathways promoting cell proliferation like E2F and G2M-checkpoint were exceptionally upregulated in TGCT and THYM and downregulated in KIRP, PRAD and LGG (see \Cref{hmap} $A$).\
The heatmap based on KEGG `GSEA` results showed four clusters as well (see \Cref{hmap} $B$).
Noticeably, O-glycan biosynthesis was highly upregulated in every single tumor type, while pathways for one-carbon metabolism, homologous recombination and cell cycle were downregulated.
The cluster with UVM, SARC, SKCM, ACC and PCPG showed high positive enrichment in metabolic pathways, like carbohydrate and drug metabolism.
Pathways associated with antigen presentation and autoimmune responses were significantly downregulated in the cluster with DLBC, LAML and KICH, as well as in CHOL and upregulated in almost all other tumor types.\
Based on the top 30 pathways according to their variance, several clusters across tumor types could be resolved (see \Cref{hm}).
The pathways included immune-, metabolic- and hallmark-associated pathways.
The highest variance overall was computed for CD8 TCR, TCR and CD8 TCR downstream pathways.
Interferon, interleukin and complement signaling pathways were often either up- or downregulated together in tumor types.
An exceptional regulation signature could be seen in THYM, where TCR- and cell-cycle-related pathways were highly upregulated.
CHOL showed a unique pattern as well with high `NES` in interferon- and inflammatory- response gene sets.

### Identification of immune signaling subtypes in 24 tumor types

```{=tex}
\begin{figure}[h]
  \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel3_Ebene 1.pdf}
  \caption{\textbf{Immune signaling subtypes identified in 24 tumor types and association with immune infiltration in KIRC} (A) Ratio of "High", "Low" and "Neutral" immune signaling expression over all patients and per tumor type (B) Heatmap showing exemplary results of clustering for KIRC (C) Heatmap showing immune infiltration in KIRC}
  \label{pengplot}
\end{figure}
```
In addition to pathway activity patterns that differ between the 33 tumor types, we next focused on assessing heterogeneity within each tumor type.
For this purpose, we used PID signaling pathways to detect whether subtypes form based on our `GSEA` results.
Hence, we identified three subtypes, "upregulated", "downregulated" and "neutral" in 24 out of 33 tumor types using `hierarchical clustering` (see \Cref{pengplot} $A$).
Significant pathways were filtered using `Kruksal-Wallis` test with an $FDR < 0.05$ and `Bonferroni` correction.
Those subtypes are only relative within one cancer type and are not representative compared to normal tissue.
The "neutral" subtype is the most common in 22 of 24 tumor types, the only exception being LUAD and PAAD, which are dominated by the "downregulated" subtype.
Generally, the "neutral" subtype is present in more than 50% of the cancer patients upon the 24 tumor types characterized by successful subtyping.
The "downregulated" and "upregulated" subtype however are approximately equally distributed.
The subtypes were particularly distinguished by CD8+ T cell receptor signaling in all 24 tumor types.

In \Cref{pengplot} $B$ an exemplary result of the subtype identification is shown for the cancer type KIRC, where the "neutral" subtype is predominant.
The top 20 pathways ranked according to their $p-value$ are displayed.
The differently expressed pathways with the highest significance are Interleukin 1 and Interleukin 27 pathways, as well as pathways related to T cell specific receptor signaling.
Beyond immune signaling pathways, cell cycle regulatory pathways including E2F, a transcription factor associated with cell proliferation, and Aurora B, which plays a role in mitosis, appear to differ between the 3 subtypes [@E2F-Yang-1995].
Therefore, the subtype with the most immune-cell related expression patterns shows the highest upregulation in cell cycle promoting pathways, characterizing this subtype as high proliferating.
An analogous correlation appears in the "downregulated" and "neutral" subtype.

Based on PID pathway activities, we conducted a more in-depth analysis of the TME in KIRC.
Therefore, we performed immune deconvolution to assess the total amount of immune cells present in tumor tissue (see \Cref{pengplot} $C$).
Patients in the "upregulated" subtype could be mapped to high CD8+ T-cell infiltration, whereas macrophages M2 and neutrophiles showed lower infiltration compared to the other two subtypes.
Nonetheless, it is visible that other types of immune cells are outstandingly present in the TMEs of the "neutral" and "downregulated" subtype.
The latter subtype showed highest infiltration rates in neutrophils, whereas the "neutral" subtype is characterized by high macrophage, as well as medium neutrophile infiltration.
Other immune cells such as NK cells and B cells were not detectable upon all subtypes.
Consequently, the subtypes differ not in immune infiltration in general, as the pathways expression patterns suggest, but are characterized by the presence of different immune cell types.

## Focused analysis on KIRC

### Characterization of metabolic and immune signaling landscape in KIRC

```{=tex}
\begin{figure}[h]
 \begin{center}
   \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel4_v2_Ebene 1.pdf}
  \end{center}
  \caption{\textbf{Identification of differences between healthy and tumorous tissue in KIRC } Heatmap showing differentially expressed hallmark pathways}
  \label{volcano}
\end{figure}
```
After characterizing differences within KIRC patients, we next wanted to identity differentially expressed genes and pathways between tumorous and healthy tissue.
Using the binary logarithm of the mean Fold change between both tissue types, we encountered that the tumorous tissue features both up and down regulated genes (see \Cref{volcanoap} $A$).
To further analyze which parameters are crucial in distinguishing tumorous from healthy tissue, we used `GSEA` on 3 gene sets: PID, KEGG, and Hallmark.

The most frequently enriched pathways, which showed differential expression in at least 50 of 72 patients, were mapped to allograft rejection, interferon gamma response and cell cycle regulators, such as E2F, G2M and KRAS.
Consequently, both immune response and cell cycle regulators appear to play an important role in tumor development and hence occur in the majority of KIRC patients.
Noticeably, all oncogenes besides the KRAS pathway are upregulated (see \Cref{volcanoap} $B$).

#### Cancer hallmark pathways form subclusters in KIRC

To characterize subtypes within the 72 KIRC patients, we next looked at each gene set individually.
We categorized the hallmark pathways into three subclusters, where certain pathways are either upregulated, downregulated or not significantly differentially expressed at all (see \Cref{volcano} $C$).
The downregulated subtype was mainly mapped to fatty acid metabolism, whereas the upregulated subtype consists of cell cycle regulatory pathways as well as immune signaling pathways.
Nonetheless, the expression patterns within these 3 subtypes of hallmark gene sets are not homogenously distributed upon all KIRC patients.
Using `hierarchical clustering`, two clusters could be identified, one smaller cluster, which shows expression levels close two zero, and one bigger cluster with highly fluctuating expression patterns.

### Identification of two KIRC subtypes that differ in pathways related to immune signaling and cell cycle regulators

```{=tex}
\begin{figure}[h]
  \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel1_Ebene 1.pdf}
  \caption{\textbf{Identifying Immune subtypes within KIRC }(A) k-means clustering of KIRC based on PID pathways (B) Heatmap showing differentially expressed pathways between both clusters (C) Proportion of "Tumor free" patients within each PID cluster (D) Fractions of CD8+ infiltration within each PID cluster}
  \label{pid}
\end{figure}
```
To determine signaling pathways that distinguish KIRC tumor tissue from normal tissue, PID pathways were analyzed in more detail.
`GSEA` revealed that significantly enriched pathways are mainly upregulated and indicated sample subclusters (see \Cref{tuk} $A$).
`k-means` clustering based on `UMAP` allowed identification of two distinct clusters (see \Cref{pid} $A$).Next, `Wilcoxon` ranked-sum test with `Bonferroni` correction determined 14 pathways that significantly differ between the two clusters, hence are crucial for formation of these two clusters.\Cref{pid} $B$ shows distinct differences in `NES` scores of these pathways in both clusters and indicate an "upregulated" and "neutral" subtype.
These differences include pathways that play an important role in cell cycle regulation, such as E2F and P53 pathway [@p53-and-E2f-Poalger-2009].
Noticeably, a majority of significantly differently pathways are related to immune response as interleukin pathways (e.g. IL1, IL2) or T cell specific receptor signaling.
More precisely, top significant pathways refer to CD8+ T cells, thus potentially differential numbers of these cells between both clusters.
This lead was confirmed by immune deconvolution, where the "upregulated" subtype is associated with higher CD8+ T cell fractions (see \Cref{pid} $D$).
Consequently, the "upregulated" cluster shows higher CD8+ T cell infiltration compared to the "neutral" subtype characterizing two different TMEs.\
Additional patient data was provided, such as gender and tumor stage and was examined with regard to PID clusters.
Only a correlation between PID subtypes and tumor stage could be observed, as $70.0\%$ of patients in the "upregulated" cluster are tumor free, while in the "neutral" cluster, only $28.1\%$ of patients characterize as tumor free (see \Cref{pid} $C$).

#### Immune infiltration subtypes can be projected to KEGG signaling and metabolic pathways

```{=tex}
\begin{figure}[h]
  \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel2_Ebene 1.pdf}
  \caption{\textbf{Identifying Metabolic and Signaling subtypes within KIRC} (A) k-means clustering based on KEGG pathways (B) Mapping of PID clusters onto KEGG clusters}
  \label{kegg}
\end{figure}
```
Next, KEGG pathways were investigated more precisely.
`GSEA` shows that enriched KEGG pathways in tumor compared to normal tissue were both, up- and downregulated (see \Cref{tuk} $B$).
Based on `UMAP`, patients could be divided into three subtypes by `k-means` clustering.
Pathways, that are significantly different between clusters, were determined using `Kruksal-Wallis` test with `Bonferroni` correction.
This allowed identification of a "fluctuating" cluster showing both up- and downregulation, one "upregulated" cluster with high or neutral `NES` and one "neutral" cluster mainly without pathway enrichment (see \Cref{kegg} $A$).
KEGG pathways, that are upregulated in the "fluctuating" and "upregulated" subtypes, but not the in "neutral" subtype, include pathways that are mainly associated with immunological processes but also cell cycle.
In this context, immunological processes refer to disease related pathways, such as diabetes mellitus, as well as cytokine and chemokine signaling and antigen processing and presentation.\
Moreover, pathways that are downregulated in the "fluctuating" subtype but neutral in both other subtypes relate mostly to metabolic pathways, such as fatty acid, amino acid, and drug metabolism, as well as TCA cycle.
Moreover, KEGG clusters show similar subdivision of samples as PID clusters.
In \Cref{kegg} $B$ it can be seen that the "fluctuating" KEGG cluster mainly matches patients of the "upregulated" PID cluster, while the "neutral" KEGG subtype includes mostly patients of the "neutral" PID subtype.
Additionally, the "upregulated" KEGG cluster corresponds to both PID clusters.

## Binary logistic regression model to predict immune infiltration based on pathway activity

The binary logistic regression model predicts immune infiltration in patients based on the pathway activity (`NES`) of `PID_TCR_PATHWAY` and `PID_IL1_PATHWAY`.
Both independent variables significantly contribute to the model as can be seen in Table \Cref{tab}, which depicts the model properties.

```{=tex}
\begin{table}[ht]
 \begin{tabular}{ |p{4.5cm}||p{3cm}|p{3cm}|p{3cm}|  }
  \hline
  \multicolumn{4}{|c|}{Logistic regression model results} \\
  \hline
   & \textbf{Estimate} & \textbf{Standard error} & \textbf{p-value}\\
  \hline
  Intercept   &  $-2.5856$  & $0.2059$ &   $<2e-16$\\
  PID\_TCR\_PATHWAY &   $0.9727$  & $0.2027$   & $1.59e-06$\\
  PID\_IL1\_PATHWAY & $1.1651$ & $0.1934$ &  $1.70e-09$\\
 \hline
 \end{tabular}
 \label{tab}
\end{table}
```
```{=tex}
\begin{figure}[t]
  \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Panel7_version3_Ebene 1.pdf}
  \caption{Logistic regression for prediction of immune infiltration (A) Regression model for $PID_TCR_PATHWAY$ and $PID_IL1_PATHWAY$ (B) Confusion Table (C) ROC-curve for model performance}
  \label{log}
\end{figure}
```
By taking the exponent of the coefficients in Table \Cref{tab}, the returned value implies that for every increase of $1$ in `NES` of the TCR and the IL-1 pathway, the odds-ratio for a patient being immune infiltrated increases by the factor $2.64517$ ($164.5\%$) and $3.206149$ ($220.6\%$), respectively.
The logistic function plots of each independent variable show that a higher enrichment score indicates a higher probability of a patient being classified as "Immune infiltrated" (see \Cref{log} $A$).
The contribution of each coefficient to the model was evaluated and confirmed with a `Wald’s` test, where both TCR ($p= 1.590e-06$) and IL-1 pathway ($p= 1.701e-09$) were significant.
The significance of the whole model in contrast to a null-model was confirmed by McFadden's `pseudo-$R^2$` ($R_{Mc}^2=0.5074508$) and the p-value of the overall `chi-square` statistic ($p=7.860549e-60$).\
The model was tested on KIRC patients in `DF2` and classified $42$ out of $72$ patients as "Immune infiltrated".
The accuracy of the model was verified by the clustering of the patients according to PID pathway activity, which showed two distinct clusters.
One cluster showed upregulation in immune pathways hence all these patients were assigned to the immune infiltrated group.
In $94.4$% of the patients the prediction and the cluster assignment corresponded correctly.
Moreover, the performance of the model was evaluated with a confusion table that compares predictions and immune infiltration according to the cluster as well.
It revealed a false-negative rate of $2.5$%, a false-positive rate of $9.4$% and a true-positive rate (recall) of $97.5$% (see \Cref{log} $B$).
The ROC-curve that was plotted for the model \Cref{log}$C$ showed an AUC of $0.9789062$.

<!-- Discussion -->

# Discussion

Cancer hallmarks define abnormalities in metabolism, immune infiltration, and cell proliferation regardless of the cancer type [@cancer_hallmarks_hanahan_2011].
Since all these aspects are rather complex and (de-)regulation takes place on transcription as well as (post-) translational levels, we tried to assess whether bulk sequencing data allow us to observe these alterations.
Therefore, we utilized 3 different gene sets, more precisely hallmark, PID and KEGG.
Notably, even though tumor pathology differs strongly among the 33 tumor types of interest, tumors always tend to cluster based on the aspects of metabolism and immune signaling.
Contrary to expectations, many essential hallmark gene sets, such as angiogenesis were not enriched throughout the tumor types at all [@transcriptomic-landscape-2017].
Notably high E2F upregulation was observed in TGCT and THYM, which is associated with cell transformation and beneficial for both tumor cell proliferation, as well as immune cell proliferation in the TME [@E2F-Yang-1995].
However, downregulation of E2F in KIRP could not be confirmed with literature [@E2F-Wang-2020].

To gain deeper insights into tumor metabolism, KEGG pathways revealed upregulation of O-glycan biosynthesis in all 33 tumor types.
This pathway is crucial in cancer cell transformation and therefore is representative for metabolic alterations [@Oglycan-Brockhausen-1999].
Beyond that higher drug metabolism in some cancer types could imply higher drug resistance and selective advantages for tumor cells [@Drug-metabolism-Michael-2005], increasing the necessity for alternative treatment options.
Downregulation of antigen presentation in some cancer types is consistent with immune evasion according to literature [@MHC-Cornel-2020] Following the lead, we used PID signaling pathways to examine the influence of immune cells in the TME.
Hence, subclusters based on PID signaling pathways were identified within each tumor type and mapped to immune cell related expression patterns in 24 out of 33 tumor types.
These tumor types were divided in "upregulated", "downregulated" and "neutral" subtypes.

To determine the foundation of the subtypes, we looked at KIRC in more detail.
Utilizing immune deconvolution on KIRC patients, we were able to assign the "upregulated" subtype to high CD8+ T cell infiltration, the "downregulated" subtype to high neutrophile infiltration and the "neutral" type to only medium immune infiltration.
These characteristics can be mapped to different compositions of the TME during tumor progression.
Early stage KIRC is often associated with macrophage M2 infiltration, also known as tumor-associated macrophages, which create an immunosuppressive microenvironment.
This tumor stage is equivalent to the "neutral" type we discovered in KIRC.
KIRC progression, however, is characterized by increased CD8+ T cell infiltration, which we observed in the "upregulated cluster" [@LandscapeKIRC-Zheng-2021].
Considering the coexistence of macrophages in this cluster, which are known for expression of immune checkpoints like PDL-1, these T cells are most likely to be exhausted, losing their anti-tumorous capacity [@TAM-Gordon-2017].
Contrary to most tumor types, macrophage and T cell infiltration correlate with poor prognosis in KIRC[@Prognosis-Fridman-2017].
The "downregulated" subtype we discovered was mainly characterized by high neutrophil expression.
Neutrophil expression in KIRC was associated with lung metastasis and therefore poor prognosis as well[@Neutrohpil-ML-Nishida-2020].

Since immune infiltration and the composition of the TME fluctuates among tumor types, no general conclusion regarding prognosis can be drawn from the 24 immune signaling pathways.
Further analysis on the immune cells present in each tumor type using immune deconvolution remains to be performed.
Additionally, Z-scoring across one gene for all samples as a substituting method for the fold change might have synthetically produced 3 subtypes.
Nonetheless, immune deconvolution, performed on the TPM values was used to confirm parts of our results.
The strict cutoff (`FDR`: 5%) used for filtering the `GSEA` results is another factor, which led to rather sparse pathway activity matrices might have created the "neutral" subtype as well.
Consequently, mean pathway expression values could have been distorted, either creating expression pattern clusters between the tumor types or cause them to vanish.

Beyond the relative differences within KIRC, tumorous tissue was compared with healthy tissue.
Noticeably, allograft rejection and interferon gamma response as well as cell cycle regulator upregulation were the most commonly enriched pathways, distinguishing tumorous from healthy tissue.
These pathways characterize an aggressive and often metastatic KIRC subtype [@Allograft-E2F-Lin-2022].
Since these hallmark pathways are upregulated in the majority of KIRC patients, the dominance of this subtype is standing to reason and is supported by further analysis.
On the first look KRAS appeared to be downregulated, which is rather untypical for an oncogene, especially since KRAS is effected in the majority of tumors.
The `Hallmark KRAS signaling DN` pathway, however, consists of genes, which are downregulated in presence of KRAS activation, which is concurrent to literature [@RAS-Kumar-2018].
Even though no mutational analysis was performed, this pathway hints at a loss of function mutation in most KIRC patients.

Further analysis, based on PID signaling pathways, revealed an "upregulated" and "neutral" subtype.
The "upregulated" subtype was mapped to high CD8+ T cell infiltration.
Additionally, this subtype is characterized by high IL1-expression, which is known to create an inflammatory environment, beneficial for tumor progression and proangiogenic [@IL1_Manovani_2017].
These subtypes were commonly found in analysis of the KIRC TME, classifying the infiltrated subtype as immunologically "hot" and the other one as "cold".
Once again, the immunologically "cold" subtype was associated with better prognosis[@mRNA-Xu-2021].
Nonetheless, the T cell infiltrated subtype showed an increased number of "Tumor free" patients.
Since therapeutic procedures are not defined for most patients, additional patient data are needed to be assess, whether these observations are truly contradictory or not.
Besides immune signaling pathways, the "upregulated" subtype showed high expression in cell cycle regulators, such as E2F.
Considering an association between the IL PI3K pathways and activation of E2F in T cells, proliferation might not only be upregulated in the tumor itself, but in T cells especially, leading to high infiltration rates[@Il2-E2F-Brennan-1997].
Furthermore, immune signaling subtypes were mapped onto metabolic and signaling KEGG clusters.
Noticeably, TCA cycle was downregulated in most patients.
This observation is associated with the Warburg effect, where glycolysis is used as a main energy source.
This leads to increased lactate production and release, creating a low pH-value, which is known to be immunosuppressive.
Additionally, the risk for apoptosis due to lesser mitochondria presence is beneficial for tumor growth[@Warburg-Zheng-2021, @Warburg-Liberti-2016].

Lastly, a tumor regression model was built to determine, whether a random KIRC patient shows expression patterns of the infiltrated or not infiltrated subtypes based on the expression of `PID_TCR_PATHWAY` and `PID_IL1_PATHWAY`.
This led to a very significant model confirmed by McFadden's `$R^2$` (0.51), which indicates an "excellent model" with values between 0.2-0.4 [@McFadden-1977].
In the training data frame (`DF1`) medium and low expression patterns were merged and thus no differentiation between "neutral" and "downregulated" considered.
Accuracy of prediction was compared to cluster number.
Since clustering was based on PID pathways, some patients could be clustered to "immune infiltrated" due to differential expression of cell cycle regulators etc.
However, both factors did not appear to lower the significance of the model.
Training on bigger sample size (here: 531) could significantly improve the model either, lowering both FPR and FNR.

Challenges, we encountered in both pan-cancer and KIRC-specific analysis, were sparse pathway activity matrices as described above.
Further, bulk sequencing data made it hard to map certain pathway activities to the cells present in the TME.
Next steps in the analysis would be to define the coherence between immune system and metabolic pathways in KIRC more precisely and project these results on our pan-cancer analysis, ideally allowing us to characterize the 24 immune signaling subtypes in depth.
Besides sequencing data, mutational analysis would be needed to gain a deeper insight into the genetic alterations associated with tumor progression.
Correlations between survival could enhance both reliability and significance of our analysis as well.

# Outlook 

Immune signaling pathway panels could be broadly introduced in cancer diagnostics and are already used for reliable subtyping.
Based on these results, the predicted immune infiltration could be used for therapeutic application, since the immune infiltrated subtype is known to be more perceivable to Immunotherapies such as Immune-Checkpoint-inhibitors [@Therapy-Chirag-2021].
This could lead to more successful treatments in early stages of the cancer.
Moreover, using mutational analysis and proteomics, neopeptide antigens could be identified.
In case of broadly similar antigens, mRNA vaccines are standing to reason, which would be especially beneficial for the immunologically "cold" phenotype, offering possible therapeutic procedures, individual for each subtype and adding prophylactic treatments as the goal of cancer treatment[@mRNA-Xu-2021].

# References

::: {#refs}
:::

# Appendix

## Supplementary figures

```{=tex}
\begin{figure}[h]
  \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/AppendixPanel6_Ebene 1.pdf}
  \caption{\textbf{Supplement: `GSEA` results for 33 tumor types} (A) Heatmap showing enrichment of hallmark gene sets in all tumor types (B) Heatmap showing enrichment of KEGG gene sets in all tumor types}
  \label{hmap}
\end{figure}
```
```{=tex}
\begin{figure}[h]
 \begin{center}
   \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Appendix_Panel4_Ebene 1.pdf}
  \end{center}
  \caption{\textbf{Supplement: Identification of differences between healthy and tumorous tissue in KIRC } (A) Volcano plot showing differential gene expression between healthy and tumorous tissue (B) Top 10 most frequently enriched pathways}
  \label{volcanoap}
\end{figure}
```
```{=tex}
\begin{figure}[h]
 \begin{center}
   \includegraphics{~/Desktop/GitHub/2022-topic-02-team-02/09_figures/Appendix_tuktuktuk_Ebene 1.pdf}
  \end{center}
  \caption{\textbf{Supplement: GSEA results for PID and KEGG pathways in DF2 } (A) } (A) Heatmap showing ES for PID pathways (B) Heatmap showing ES for KEGG pathways
  \label{tuk}
\end{figure}
```


## Packages
\begin{table}[!ht]
    \centering
    \begin{tabular}{|l|l|l|}
    \hline
        Package & Author & Application \\ \hline
        biomaRt & [@BioMart] & biotype filtering  \\ \hline
        circlize & [@circlize] & color scaling for heatmaps  \\ \hline
        ComplexHeatmap & [@ComplexHeatmap] & heatmap plots  \\ \hline
        dplyr & [@dplyr] & data transformation  \\ \hline
        factoextra & [@factoextra] & visualization of k-means results  \\ \hline
        fgsea & [@fgsea] & curation of pathway activity matrix  \\ \hline
        ggdendro & [@ggdendro] & hierarchical clustering  \\ \hline
        ggplot2 & [@ggplot2] & plots  \\ \hline
        ggpubr & [@ggpubr] & r - optimization of heatmap plots  \\ \hline
        grid & [@grid] & creation of plot panels  \\ \hline
        msigdbr & [@msigdbr] & curration of pathways  \\ \hline
        reshape2 & [@reshape2] & data preperation for ggplot  \\ \hline
        stats & [@stats] & statistical testing  \\ \hline
        stringr & [@stringr] & text formatting  \\ \hline
        tidyverse & [@tidyverse] & data transformation  \\ \hline
        umap & [@umap] & dimension reduction, visualisation of PCA results  \\ \hline
        uwot & [@uwot] & UMAP, dimension reduction  \\ \hline
    \end{tabular}
\end{table}
