---
title: " "
output:
  pdf_document:
    number_sections: TRUE
  html_document:
    code_folding: hide
documentclass: scrreprt
classoption: [parskip, oneside] # für drucken twoside
header-includes:
  - \usepackage[ ngerman, main=english]{babel}
  - \usepackage[utf8]{inputenc}
  - \usepackage[T1]{fontenc}
  - \usepackage{lmodern}
  - \usepackage[onehalfspacing]{setspace}
  - \usepackage[left=2.50cm, right=2.50cm, top=2.50cm, bottom=2.50cm, bindingoffset=10mm, includehead, includefoot]{geometry}
  - \usepackage[headsepline]{scrlayer-scrpage}
  - \usepackage{url}
  - \usepackage[backend=biber, style=authoryear, giveninits=true, maxbibnames=99, uniquename=init, maxcitenames=2, hyperref=true, date=year]{biblatex}
  - \usepackage{xpatch}
  - \usepackage{csquotes}
  - \usepackage{amsmath}
  - \usepackage{listings}
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{multirow}
  - \usepackage{rotating}
  - \usepackage{subfigure}
  - \usepackage{graphicx}
  - \usepackage{float}
  - \usepackage{acronym}
  - \usepackage{lipsum}
  - \usepackage{scrhack}
  - \emergencystretch=50pt
  - \clubpenalty = 10000
  - \widowpenalty = 10000
  - \displaywidowpenalty = 10000
  - \automark[section]{chapter}
  - \renewcommand*{\chaptermarkformat}{}
  - \renewcommand*{\sectionmarkformat}{}
  - \setkomafont{title}{\sffamily}
  - \setkomafont{disposition}{\usekomafont{title}}
  - \setkomafont{author}{\usekomafont{title}}
  - \setkomafont{date}{\usekomafont{title}}
  - \setkomafont{caption}{\sffamily\small}
  - \setkomafont{captionlabel}{\usekomafont{caption}\bfseries\small}
  - \setkomafont{pagehead}{\normalfont\scshape}
csl: bib/molecular-biology-of-the-cell.csl
bibliography: [bib/packages.bib, bib/BibTeX.bib]
---
<!-- Based on Nicolas Peschkes LaTeX Template for MoBis, if you want further information on the packages just check it here: https://gitlab.com/npeschke/LaTeX-Template-IPMB -->

<!-- your title page -->

\begin{titlepage}
\centering
	{\Large Ruprecht-Karls-Universität Heidelberg\\
		Fakultät für Biowissenschaften\\
		Bachelorstudiengang Molekulare Biotechnologie\\}

	{\vspace{\stretch{2}}}
	{\usekomafont{title}

		{\Huge ksdflsdjf}

		{\Huge sfdsfd}

		{\Huge sfsf}

	}

	\vspace{\stretch{2}}
	{\Large Data Science Project SoSe 2022}

	\vspace{\stretch{2}}

	{\Large
		\begin{tabular}{rl}
			Autoreb & Max Mustermann, jklsdfksldkfslkdjf\\
			Geburtsort & sdfjksafdls\\
			Abgabetermin &20.07.2022\\
		\end{tabular}
	}

	\vspace{\stretch{1}}

\end{titlepage}


<!-- add your chapters here -->
\tableofcontents

<!-- Intoduction -->
# Introduction


## Computational Tools

### Dimension reduction

#### PCA

#### UMAP 


### Statistical analysis

#### Shapiro-Wilks test
Shapiro-Wilks (SW) test is a normality test based on regression and correlation. It tests the null hypothesis that the data follows a normal distribution. Small values of SW test statistic indicate no normality of the data thus the null hypothesis is rejected. SW values of one suggest normality @Shapiro-Yap-2011.

#### Wilcoxon rank-sum and signed-rank test
Wilcoxon rank-sum test and Wilcoxon signed-rank test both are non-parametric statistical hypothesis tests that can be used when the data does not follow a normal distribution.  Wilcoxon signed-rank test is used to analyze matched-pair or one-sample data. It tests the null hypothesis that there is no difference in probability distribution of first and second sample, hence the distribution of pairwise differences is centered at zero. The test is based on ranked absolute values of differences @Wilcoxon-Woolson-2007. Wilcoxon rank-sum test is performed when analyzing unpaired-data and is likewise based on ranked values. The null hypothesis states that there is no association between the two samples @Wilcoxon-Rey2011. 

#### Kruksal-Wallis 
Kruksal-Wallis test is a rank-based non-parametric hypothesis test. It is an extension of Wilcoxon rank-sum test and can allows comparing more than two independent data sets. Kruksal-Wallis test tests the null hypothesis that there is no difference in distributions of all $k$ data sets. Alternative hypothesis states that there is a difference in at least two populations **(oder: …that at least two populations show stochastic heterogeneity)** @Kruksal-Wallis-ostertagova-2014. 

#### Bonferroni correction
Multiple statistical testing results in an increased false positive rate (type I errors). Bonferroni correction is used to reduce this type I error rate. For this, the significance level is adjusted by dividing the critical $p$-value $\alpha$ by the number of tests @Bonferroni-Amstrong-2014. 

### Clustering

#### Kmeans

#### Hierarchial clustering

### Immune deconvolution

The immune deconvolution package @immunedeconv-merotto-2022 is used to obtain immune cell fractions from bulk RNA-sequencing data. The input is a matrix with genes as rows and samples as columns, containing gene expression values as transcripts per million (TPM). The deconvolution algorithm models the expression of a single gene as a linear combination of the expression of that gene across the different cell types. An equation for each gene is set up that contains terms of the matrix multiplication of a signature matrix $S$ and an immune cell fraction vector $F$. The signature matrix $S$ contains all average gene expression profiles for each gene in the immune cell types, respectively. The output using the “quanTIseq” method is a matrix containing the immune cell fractions for each sample that have been re-normalized to sum up to one @Immune-deconvolution-Finotello-2018.

### GSEA


### Regression

<!-- Materials and Methods -->

# Materials and Methods

## Data cleaning

The analysis was focused on two data sets. Pan-cancer analysis was performed on gene expression data of 9783 patients of 33 tumor types (Pan-cancer data set (DS)). Focused analysis was based on gene expression data of normal and tumor tissue of 72 KIRC patients (KIRC DS). The data obtained was already normalized by $log2$(Transcripts per million).
Both data sets were filtered for protein-coding genes using biomaRt package. This reduced gene expression data in pan cancer DS from 60 498 to 19 624 genes and in KIRC DS from 19624 to 19186 genes. In the pan-cancer DS, variance was computed for each gene over all samples and the lower p50-quantile was subsequently removed. Furthermore, constantly expressed genes only were removed in KIRC DS, resulting in gene expression data of 18 645 genes.

## TCGA pan-cancer analysis

## Focused Analysis: Identifying subtypes in KIRC
For focused analysis, KIRC was examined in more detail.  The analysis was based on gene expression data of normal and tumor tissue of 72 patients. Data cleaning was performed as described above.\\
First, differential gene expression in tumor tissue compared to normal tissue was analyzed by calculation of Foldchange $FC =(mean condition 1) / (mean condition 2)$ for each gene, where condition one represents gene expression of tumor tissue and condition two gene expression of normal tissue. Statistical significance was determined using Wilcoxon test with Bonferroni correction, as Shapiro-Wilks test indicated no normality of the data.
The second part of this analysis was focused on differential pathway activity. Pathway activity matrices were determined using GSEA, where genes were ranked based on their foldchange. Pathways with an adjusted $p$-value $>0.05$ were considered as non-significant, hence their NES was set to zero. Hallmark-pathways, KEGG-pathways and PID-pathways were analyzed. Patients were compared based on their pathway activity. Subclusters of patients were visualized for each geneset by UMAP calculated on PCA-results and subsequently identified by k-means clustering. Optimal number of clusters was determined using elbow method and silhouette method. **Differences in pathway activity significantly defining these clusters were detected using Wilcoxon test for two clusters and H-test for three clusters, both with Bonferroni correction (?!)--> umformulieren**. 

Pathways that were crucial for this clustering and were significantly different between those clusters were detected using Wilcoxon test for two clusters and H-test for three clusters, both with Bonferroni correction.

## Packages

<!-- Results -->

# Results

## TCGA pan-cancer analysis

## KIRC specific analysis

## Logistic regression 

\begin{tabular}{ |p{4.5cm}||p{3cm}|p{3cm}|p{3cm}|  }
 \hline
 \multicolumn{4}{|c|}{Logistic regression model results} \\
 \hline
  & \textbf{Estimate} & \textbf{Standard error} & \textbf{p-value}\\
 \hline
 Intercept   &  $0.18303$  & $0.01336$ &   $<2e-16$\\
 PID\_TCR\_PATHWAY &   $0.09727$  & $0.01844$   & $1.95e-07$\\
 PID\_IL1\_PATHWAY & $0.12211$ & $0.01801$ &  $3.23e-11$\\
\hline
\end{tabular}


<!-- Discussion -->
# Discussion

<!-- Concluding remarks/Outlook -->

# Concluding remarks/Outlook




# References

<div id="refs"></div>

# Appendix
