---
title: "Metabolic_Pathways_Anna"
author: "Anna von Bachmann"
date: '2022-06-08'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(msigdbr)
```

```{r}
# Obtaining gene sets from msigdb for Carbohydrate metabolism
Homo_sapiensCPR = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CP:REACTOME")
Carbohydrates = Homo_sapiensCPR[Homo_sapiensCPR$gs_id=="M16864",]

# Glycolysis
Homo_sapiensCGP = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CGP")
Glycolysis = Homo_sapiensCGP[Homo_sapiensCGP$gs_id=="M16111",] #22 genes

# Amino acid 
Homo_sapiensC2 = msigdbr(species = "Homo sapiens", category = "C2")
Amino_Acid = Homo_sapiensC2[Homo_sapiensC2$gs_id=="M39570",] #102 genes

#Sphingolipid
Sphingolipid = Homo_sapiensCPR[Homo_sapiensCPR$gs_id=="M14857",]

#Fatty acid
Homo_sapiensCPK = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CP:KEGG")
Fatty_acid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M699",]

#Pentose phosphate
Pentose_phosphate = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M1386",]

#Ether lipid
Ether_lipid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M2130",]

#Glutathione
Glutathione = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M1840",]

#Nitrogen
Nitrogen = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M4629",]

#Oxidative phosphorylation
Oxidative_phosphorylation = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M19540",]

#Purines
Purine = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M14314",]

#Pyrimidines
Pyrimidine = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M5109",]

#Pyruvate
Pyruvate = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M7934",]

#Steroids
Steroid = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M5872",]

#Steroid hormones
Steroid_hormones = Homo_sapiensCPK[Homo_sapiensCPK$gs_id=="M14933",]
```

```{r}
#create GSEA compatible gene set list, NCBI entrez gene id as characters
carbohydrates = as.character(Carbohydrates$ensembl_gene)
glycolysis = as.character(Glycolysis$ensembl_gene)
amino_acid = as.character(Amino_Acid$ensembl_gene)
sphingolipid = as.character(Sphingolipid$ensembl_gene)
fatty_acid = as.character(Fatty_acid$ensembl_gene)
pentose_phosphate = as.character(Pentose_phosphate$ensembl_gene)
ether_lipid = as.character(Ether_lipid$ensembl_gene)
glutathione = as.character(Glutathione$ensembl_gene)
nitrogen = as.character(Nitrogen$ensembl_gene)
oxidative_phosphorylation = as.character(Oxidative_phosphorylation$ensembl_gene)
purine = as.character(Purine$ensembl_gene)
pyrimidine = as.character(Pyrimidine$ensembl_gene)
pyruvate = as.character(Pyruvate$ensembl_gene)
steroid = as.character(Steroid$ensembl_gene)
steroid_hormones = as.character(Steroid_hormones$ensembl_gene)

met.pathways = list(carbohydrates, glycolysis, amino_acid, sphingolipid, fatty_acid, pentose_phosphate, ether_lipid, glutathione, nitrogen, oxidative_phosphorylation, purine, pyrimidine, pyruvate, steroid, steroid_hormones)

names(met.pathways) = c("M16864_Carbohydrates", "M16111_Glycolysis", "M39570_Amino_acid", "M14857_Sphingolipid", "M699_Fatty_Acid", "M1386_Pentose_phosphate", "M2130_Ether_lipid", "M1840_Glutathione", "M4629_Nitrogen", "M19540_Oxidative_phosphorylation", "M14314_Purine", "M5109_Pyrimidine", "M7934_Pyruvate", "M5872_Steroid", "M14933_Steroid_hormones")
```


```{r} 

#next adapt format to hallmark genesets and extract the genes from our filtered matrix
carbohydrates_genesymbols = Carbohydrates$human_gene_symbol
glycolysis_genesymbols = Glycolysis$human_gene_symbol
amino_Acid_genesymbols = Amino_Acid$human_gene_symbol
nucleotide_genesymbols = Nucleotide$human_gene_symbol

metabolic_pathways = list(
  carbohydrate = carbohydrates_genesymbols,
  glycolysis = glycolysis_genesymbols,
  amino_acid = amino_Acid_genesymbols
  )
```

```{r}
#load metabolic genesets
hallmarks = msigdbr(species = "Homo sapiens", category = "H")
#create a df with metabolic pathways and genes as columns 
hallmarks_gsname = hallmarks$gs_name
hallmarks_ensembl = hallmarks$ensembl_gene
hallmarks = as.data.frame(cbind(hallmarks_gsname, hallmarks_ensembl))

#create a df with names of pathways
pathway_names = c(unique(hallmarks$hallmarks_gsname))
#create a list in which the pathways will be stored (outout has to be a list)
metabolic_genesets = list(list())#
#fill the list with pathways and genes
for (x in 1:length(pathway_names)){
  metabolic_genesets[[x]] = pathway_names[x]
  z = c((hallmarks[which(hallmarks$hallmarks_gsname == pathway_names[x]),2]))
  for (k in 1:length(z)){
  metabolic_genesets[[x]][[k]] = z[k]
  }}
names(metabolic_genesets) = pathway_names
```

```{r}
#additional PID pathways (signaling)
PID_pathways = msigdbr(species = "Homo sapiens", category = "C2", subcategory  = "CP:PID")

#create a df with metabolic pathways and genes as columns 
PID_gsname = PID_pathways$gs_name
PID_ensembl = PID_pathways$ensembl_gene
PID_pathways = as.data.frame(cbind(PID_gsname, PID_ensembl))

#create a vector with names of pathways
PID_pathway_names = c(unique(PID_pathways$PID_gsname))
#create a list in which the pathways will be stored (outout has to be a list)
PID_genesets = list(list())
#fill the list with pathways and genes
for (x in 1:length(PID_pathway_names)){
  PID_genesets[[x]] = PID_pathway_names[x]
  z = c((PID_pathways[which(PID_pathways$PID_gsname == PID_pathway_names[x]),2]))
  for (k in 1:length(z)){
  PID_genesets[[x]][[k]] = z[k]
  }}
names(PID_genesets) = PID_pathway_names
```

