---
title: "Logistic regression"
author: "Maja Glotz"
date: "6/28/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(69)
```

```{r load packages}
library(immunedeconv)
library("org.Hs.eg.db")
library(gplots)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(rlist)
library(stats)
```

```{r}
set.seed(69)
#check immune infiltration in KIRC tumor dataset
tcga_KIRC_tumor_cl <- readRDS("../data/tcga_KIRC_tumor_cl.rds")
#convert ensemble into gene symbol
ensemblids = rownames(tcga_KIRC_tumor_cl)
symbols = mapIds(org.Hs.eg.db, keys = ensemblids, keytype = "ENSEMBL", column="SYMBOL")
rownames(tcga_KIRC_tumor_cl) = symbols
KIRC_tumor = as.matrix(tcga_KIRC_tumor_cl)

#use package immune deconvolution
immune_cells = deconvolute(KIRC_tumor, "xcell")
cell_types = immune_cells$cell_type
immune_cells = immune_cells[,-1]
rownames(immune_cells) = cell_types

immune_cells_matrix = as.matrix(immune_cells)
heatmap.2(immune_cells_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

#reverse log2 operation --> Transcripts per million (TPM)
KIRC_tumor_tpm = 2^KIRC_tumor
quantiseq = deconvolute(KIRC_tumor_tpm, "quantiseq", tumor = "true")
cell_types_2 = quantiseq$cell_type
quantiseq_matrix = quantiseq[,-1]
rownames(quantiseq_matrix) = cell_types_2
quantiseq_matrix = as.matrix(quantiseq_matrix)
quantiseq_matrix = quantiseq_matrix[-c(11),]

heatmap.2(quantiseq_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

quantiseq_cluster1 = quantiseq[, c("cell_type", PID_c1_names)]
quantiseq_cluster2 = quantiseq[, c("cell_type",PID_c2_names)]

#barplot of immune infiltration QUANTISEQ
quantiseq_cluster1_plot = quantiseq_cluster1 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot = quantiseq_cluster2 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot, quantiseq_cluster2_plot, ncol = 1, nrow = 2)
#PATIENTEN NAMEN NICHT KORREKT ZUGEORDNET

#CD8 only
quantiseq_cluster1_CD8 = quantiseq_cluster1[8,]
quantiseq_cluster2_CD8 = quantiseq_cluster2[8,]
quantiseq_CD8 = cbind(quantiseq_cluster1_CD8, quantiseq_cluster2_CD8)
quantiseq_CD8 = quantiseq_CD8[, -c(42)]

sum(quantiseq_cluster1_CD8[1, 2:41]) #2.008835
sum(quantiseq_cluster2_CD8[1, 2:33]) #0.3102198 

#plot in one
quantiseq_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_CD8))) +
    theme(axis.text.y = element_text(size = 4)) #Reihenfolge bleibt nicht gleich...

#plot both separately
quantiseq_cluster1_plot_CD8 = quantiseq_cluster1_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_CD8))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot_CD8 = quantiseq_cluster2_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_CD8))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot_CD8, quantiseq_cluster2_plot_CD8, ncol = 1, nrow = 2)

tcga_tumor_annotation %>% filter_all(any_vars(. %in% c("TCGA-CW-6087-01"))) #patient with extremely high immune infiltration was tumor FREE in the end!

#T-reg only:
quantiseq_cluster1_Treg = quantiseq_cluster1[9,]
quantiseq_cluster2_Treg = quantiseq_cluster2[9,]
quantiseq_Treg = cbind(quantiseq_cluster1_CD8, quantiseq_cluster2_CD8)
quantiseq_Treg = quantiseq_Treg[, -c(42)]

sum(quantiseq_cluster1_Treg[1, 2:41]) #0.2667334
sum(quantiseq_cluster2_Treg[1, 2:33]) #0.2479281

#plot both separately
quantiseq_cluster1_plot_Treg = quantiseq_cluster1_Treg %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_Treg))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot_Treg = quantiseq_cluster2_Treg %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_Treg))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot_Treg, quantiseq_cluster2_plot_Treg, ncol = 1, nrow = 2)

#Neutrophils only:
quantiseq_cluster1_neutro = quantiseq_cluster1[5,]
quantiseq_cluster2_neutro = quantiseq_cluster2[5,]
quantiseq_neutro = cbind(quantiseq_cluster1_neutro, quantiseq_cluster2_neutro)
quantiseq_neutro = quantiseq_neutro[, -c(42)]

sum(quantiseq_cluster1_neutro[1, 2:41]) #4.812149
sum(quantiseq_cluster2_neutro[1, 2:33]) #3.784561

#plot both separately
quantiseq_cluster1_plot_neutro = quantiseq_cluster1_neutro %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_neutro))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot_neutro = quantiseq_cluster2_neutro %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_neutro))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot_neutro, quantiseq_cluster2_plot_neutro, ncol = 1, nrow = 2)


#MCP --> arbitrary values only for comparison between samples
mcp_counter = deconvolute(KIRC_tumor_tpm, "mcp_counter")
mcp_counter_matrix = as.matrix(mcp_counter)
mcp_counter_matrix = mcp_counter_matrix[-c(11),]
heatmap.2(quantiseq_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

mcp_counter %>%
  gather(sample, score, -cell_type) %>%
  ggplot(aes(x=sample, y=score, color=cell_type)) +
    geom_point(size=4) +
    facet_wrap(~cell_type, scales="free_x", ncol=3) +
    scale_color_brewer(palette="Paired", guide=FALSE) +
    coord_flip() + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```

```{r}
#just based on mean NES top up- and downregulated pathways
KIRC_PID_mean = as.data.frame(apply(KIRC_PID_cl, 1, mean))
colnames(KIRC_PID_mean) = "NES"
KIRC_PID_topup = slice_max(KIRC_PID_mean, order_by = KIRC_PID_mean[,1], n = 10)
KIRC_PID_topdown = slice_min(KIRC_PID_mean, order_by = KIRC_PID_mean[,1], n = 10)
KIRC_PID_topdown = KIRC_PID_topdown[order(nrow(KIRC_PID_topdown):1),, drop = FALSE]
KIRC_PID_top = rbind(KIRC_PID_topup, KIRC_PID_topdown)

#top pathways significantly different NES in KIRC 2 clusters
PID_sigpw = c("PID_CD8_TCR_PATHWAY", "PID_CD8_TCR_DOWNSTREAM_PATHWAY", "PID_IL12_2PATHWAY", "PID_TCR_PATHWAY", "PID_IL27_PATHWAY", "PID_IL1_PATHWAY", "PID_CXCR4_PATHWAY", "PID_FOXM1_PATHWAY", "PID_IL8_CXCR2_PATHWAY", "PID_E2F_PATHWAY")
KIRC_top_pw = KIRC_PID_cl[c(PID_sigpw),]

#compute correlation between NES of PID pathways (rowise) --> need to transpose it first
KIRC_top_pw_cor_pearson = cor(t(KIRC_top_pw), method = "pearson")
Heatmap(KIRC_top_pw_cor_pearson, name = "mat", cluster_rows = FALSE, cluster_columns = FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 10))
KIRC_top_pw_cor_spearman= cor(t(KIRC_top_pw), method = "spearman")
Heatmap(KIRC_top_pw_cor_spearman, name = "mat", cluster_rows = FALSE, cluster_columns = FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 10))

#add new row according to cluster, if in cluster 1 --> 1, if in cluster 2 --> 0
cluster_number = vector(mode = 'list',72)
KIRC_patients = as.data.frame(colnames(KIRC_top_pw))
for (i in 1:ncol(KIRC_top_pw)){
  if (KIRC_patients[i,] %in% PID_c1_names){
    cluster_number[[i]] = 1
  } else {
    cluster_number[[i]] = 0
  }
}

cluster_numbers = list.cbind(cluster_number)
colnames(cluster_numbers) = colnames(KIRC_top_pw)
KIRC_top_pw = rbind(KIRC_top_pw, cluster_numbers)
rownames(KIRC_top_pw)[rownames(KIRC_top_pw) == "1"] <- "Infiltration"
KIRC_top_pw_t = as.data.frame(t(KIRC_top_pw))

#try simple logistic test regression

# Null-Modell
model0 = glm(Infiltration ~ 1, data = KIRC_top_pw_t)
summary(model0)

# Test-Modell
model1 = glm(Infiltration ~ PID_CD8_TCR_PATHWAY + PID_TCR_PATHWAY + PID_IL27_PATHWAY + PID_CXCR4_PATHWAY + PID_FOXM1_PATHWAY, data = KIRC_top_pw_t) #nur signifikante pw genommen (Beitrag leisten)
summary(model1)

#Omnibus-Test
modelchi = model1$null.deviance - model1$deviance
chidf = model1$df.null -model1$df.residual
chisqp = 1 - pchisq(modelchi, chidf) # <0.05 test-Modell leistet signifikanten Erklärungsbeitrag

#Gütemaße
n = length(model1$residuals)
R2cs = 1-exp((model1$deviance - model1$null.deviance)/n)
R2n = R2cs/(1-exp(-(model1$null.deviance/n)))

#Predict
glm_probs = data.frame(probs = predict(model1, type="response"))
head(glm_probs)

KIRC_top_pw_predict = rbind(KIRC_top_pw, t(glm_probs))

probabilities <- model1 %>% predict(KIRC_top_pw_t, type = "response")
predicted.classes <- data.frame(ifelse(probabilities > 0.5, "1", "0"))
colnames(predicted.classes) = "predicted.classes"
KIRC_top_pw_predict = rbind(KIRC_top_pw_predict, t(predicted.classes))
#accuracy? 
mean(predicted.classes == KIRC_top_pw_t$Infiltration) #0.9583333


#Plot
KIRC_top_pw_t %>%
  mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
  ggplot(aes(PID_CD8_TCR_PATHWAY, prob)) +
  geom_point(alpha = 0.2) +
  geom_smooth(method = "glm", method.args = list(family = "binomial")) +
  labs(
    title = "Logistic Regression Model", 
    x = "PID_CD8_TCR_PATHWAY activity",
    y = "Probability of immune infiltration"
    )
```


```{r}
#try to test on KIRC data fro m TCGA (531 samples)
KIRC_tcga = pathway_activity_matrix %>%
    select(starts_with('KIRC'))
KIRC_tcga_PID = KIRC_tcga[PID_sigpw,]
KIRC_tcga_PID_t = data.frame(t(KIRC_tcga_PID))
#predict
probabilities_2 <- model1 %>% predict(KIRC_tcga_PID_t, type = "response")
predicted.classes_2 <- data.frame(ifelse(probabilities_2 > 0.5, "1", "0"))
colnames(predicted.classes_2) = "prediction"
predicted.classes_2 = t(predicted.classes_2)
sum(as.numeric(predicted.classes_2[1,])) #69 immune infiltrated patients

#create list with all patients who have a 1 in "prediction"
KIRC_tcga_immune_patients = numeric()
KIRC_tcga_PID_predict = rbind(KIRC_tcga_PID, predicted.classes_2)
for (i in 1:ncol(KIRC_tcga_PID_predict)){
  if (KIRC_tcga_PID_predict[11, i] == "1")
    KIRC_tcga_immune_patients = append(KIRC_tcga_immune_patients, colnames(KIRC_tcga_PID_predict[i]))
}
#remove first letters (KIRC.) from sample ID
KIRC_tcga_immune_patients = str_sub(KIRC_tcga_immune_patients, 6)

#get annotations
KIRC_tcga_immunep_annotations = tcga_tumor_annotation %>% filter(sample %in% KIRC_tcga_immune_patients)

#check with immune deconvolution again
KIRC_tcga_tpm = 2^tcga_exp_KIRC
ensemblids_tcga_cleaned = rownames(KIRC_tcga_tpm)
symbols_tcga_cleaned = mapIds(org.Hs.eg.db, keys = ensemblids_tcga_cleaned, keytype = "ENSEMBL", column="SYMBOL")#336 duplicated
geneIDs <- ensembldb::select(EnsDb.Hsapiens.v79, keys= ensemblids_tcga_cleaned, keytype = "GENEID", columns = c("SYMBOL","GENEID")) #only 9775 gene symbols found --> remove missing genes from data frame

#remove duplicated gene symbols (24)
geneIDs_unique = distinct(geneIDs,SYMBOL, .keep_all= TRUE)
ensemblid_found = as.vector(geneIDs_unique$GENEID)
KIRC_tcga_tpm_filtered = KIRC_tcga_tpm %>% filter(rownames(KIRC_tcga_tpm) %in% ensemblid_found)
rownames(KIRC_tcga_tpm_filtered) = geneIDs_unique$SYMBOL
KIRC_tcga_tpm_matrix = as.matrix(KIRC_tcga_tpm_filtered)

#immune deconvolution
quantiseq = deconvolute(KIRC_tcga_tpm_matrix, "quantiseq", tumor = "true")

cell_types_2 = quantiseq$cell_type
quantiseq_matrix = quantiseq[,-1]
rownames(quantiseq_matrix) = cell_types_2
quantiseq_matrix = as.matrix(quantiseq_matrix)
quantiseq_matrix = quantiseq_matrix[-c(11),]
heatmap.2(quantiseq_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))
Heatmap(quantiseq_matrix, name = "mat", column_km = 5, column_names_gp = gpar(fontsize = 2), width = unit(27, "cm"), height = unit(12, "cm"))

```

