---
title: "Logistic regression"
author: "Maja Glotz"
date: "6/28/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(69)
```

```{r load packages}
library(immunedeconv)
library("org.Hs.eg.db")
library(gplots)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(rlist)
library(stats)
library(tidyselect)
library(cvms)
library(ROCR)
```

```{r set seed}
set.seed(69)
```

```{r loading data}
tcga_KIRC_tumor_cl <- readRDS("data/tcga_KIRC_tumor_cl.rds")
PID_c1_names <- readRDS("data/PID_c1_names.rds")
PID_c2_names <- readRDS("data/PID_c2_names.rds")
c1_patients <- readRDS("/data/c1_patients.RDS")
```


```{r immune deconvolution in small KIRC data set}
#convert ensemble id into gene symbol
ensemblids = rownames(tcga_KIRC_tumor_cl)
symbols = mapIds(org.Hs.eg.db, keys = ensemblids, keytype = "ENSEMBL", column="SYMBOL")
rownames(tcga_KIRC_tumor_cl) = symbols
KIRC_tumor = as.matrix(tcga_KIRC_tumor_cl)

#use immune deconvolution package
#xcell method 
immune_cells = deconvolute(KIRC_tumor, "xcell")
cell_types = immune_cells$cell_type
immune_cells = immune_cells[,-1]
rownames(immune_cells) = cell_types

immune_cells_matrix = as.matrix(immune_cells)
heatmap.2(immune_cells_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

#reverse log2 operation --> Transcripts per million (TPM)
KIRC_tumor_tpm = 2^KIRC_tumor
#quantiseq method --> returns immune cell fractions
quantiseq = deconvolute(KIRC_tumor_tpm, "quantiseq", tumor = "true")
cell_types_2 = quantiseq$cell_type
quantiseq_matrix = quantiseq[,-1]
rownames(quantiseq_matrix) = cell_types_2
quantiseq_matrix = as.matrix(quantiseq_matrix)
quantiseq_matrix = quantiseq_matrix[-c(11),]
#plot immune cell fractions
heatmap.2(quantiseq_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

#split quantiseq matrix according to PID clustering (PCA, UMAP, k-means)
quantiseq_cluster1 = quantiseq[, c("cell_type", PID_c1_names)]
quantiseq_cluster2 = quantiseq[, c("cell_type",PID_c2_names)]

#barplot of immune infiltration QUANTISEQ
quantiseq_cluster1_plot = quantiseq_cluster1 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot = quantiseq_cluster2 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot, quantiseq_cluster2_plot, ncol = 1, nrow = 2)

#CD8 only
quantiseq_cluster1_CD8 = quantiseq_cluster1[8,]
quantiseq_cluster2_CD8 = quantiseq_cluster2[8,]
quantiseq_CD8 = cbind(quantiseq_cluster1_CD8, quantiseq_cluster2_CD8)
quantiseq_CD8 = quantiseq_CD8[, -c(42)]

sum(quantiseq_cluster1_CD8[1, 2:41]) #2.008835
sum(quantiseq_cluster2_CD8[1, 2:33]) #0.3102198 

#plot in one
quantiseq_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_CD8))) +
    theme(axis.text.y = element_text(size = 4)) #order changes, clusters mixed...

#plot both separately
quantiseq_cluster1_plot_CD8 = quantiseq_cluster1_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_manual(values = "Blue") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_CD8))) +
    theme(axis.text.y = element_text(size = 4)) 
quantiseq_cluster1_plot_CD8

quantiseq_cluster2_plot_CD8 = quantiseq_cluster2_CD8 %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_manual(values = "Blue") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_CD8))) +
    theme(axis.text.y = element_text(size = 4)) + 
    ylim(0,0.55)
quantiseq_cluster2_plot_CD8

ggarrange(quantiseq_cluster1_plot_CD8, quantiseq_cluster2_plot_CD8, ncol = 1, nrow = 2)

#"TCGA-CW-6087-01" patient with extremely high CD8+ immune infiltration!

#T-reg only:
quantiseq_cluster1_Treg = quantiseq_cluster1[9,]
quantiseq_cluster2_Treg = quantiseq_cluster2[9,]
quantiseq_Treg = cbind(quantiseq_cluster1_CD8, quantiseq_cluster2_CD8)
quantiseq_Treg = quantiseq_Treg[, -c(42)]

sum(quantiseq_cluster1_Treg[1, 2:41]) #0.2667334
sum(quantiseq_cluster2_Treg[1, 2:33]) #0.2479281

#plot both separately
quantiseq_cluster1_plot_Treg = quantiseq_cluster1_Treg %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_Treg))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot_Treg = quantiseq_cluster2_Treg %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_Treg))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot_Treg, quantiseq_cluster2_plot_Treg, ncol = 1, nrow = 2)

#Neutrophils only:
quantiseq_cluster1_neutro = quantiseq_cluster1[5,]
quantiseq_cluster2_neutro = quantiseq_cluster2[5,]
quantiseq_neutro = cbind(quantiseq_cluster1_neutro, quantiseq_cluster2_neutro)
quantiseq_neutro = quantiseq_neutro[, -c(42)]

sum(quantiseq_cluster1_neutro[1, 2:41]) #4.812149
sum(quantiseq_cluster2_neutro[1, 2:33]) #3.784561

#plot both separately
quantiseq_cluster1_plot_neutro = quantiseq_cluster1_neutro %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster1_neutro))) +
    theme(axis.text.y = element_text(size = 4)) 

quantiseq_cluster2_plot_neutro = quantiseq_cluster2_neutro %>%
  gather(sample, fraction, -cell_type) %>%
  ggplot(aes(x=sample, y=fraction, fill=cell_type)) +
    geom_bar(stat='identity') +
    coord_flip() +
    scale_fill_brewer(palette="Paired") + 
    scale_x_discrete(limits = rev(levels(quantiseq_cluster2_neutro))) +
    theme(axis.text.y = element_text(size = 4)) 

ggarrange(quantiseq_cluster1_plot_neutro, quantiseq_cluster2_plot_neutro, ncol = 1, nrow = 2)

#MCP --> arbitrary values only for comparison between samples
mcp_counter = deconvolute(KIRC_tumor_tpm, "mcp_counter")
mcp_counter_matrix = as.matrix(mcp_counter)
mcp_counter_matrix = mcp_counter_matrix[-c(11),]
heatmap.2(quantiseq_matrix, dendrogram = "column", hclustfun = hclust, trace = "none", margins = c(11,11))

mcp_counter %>%
  gather(sample, score, -cell_type) %>%
  ggplot(aes(x=sample, y=score, color=cell_type)) +
    geom_point(size=4) +
    facet_wrap(~cell_type, scales="free_x", ncol=3) +
    scale_color_brewer(palette="Paired", guide=FALSE) +
    coord_flip() + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```


```{r logistic regression model trained on small KIRC data set}
#top pathways significantly different pathways in KIRC 2 clusters
#correct???
PID_sigpw = c("PID_CD8_TCR_PATHWAY", "PID_CD8_TCR_DOWNSTREAM_PATHWAY", "PID_IL12_2PATHWAY", "PID_TCR_PATHWAY", "PID_IL27_PATHWAY", "PID_IL1_PATHWAY", "PID_CXCR4_PATHWAY", "PID_FOXM1_PATHWAY", "PID_IL8_CXCR2_PATHWAY", "PID_E2F_PATHWAY")
KIRC_top_pw = KIRC_PID_cl[c(PID_sigpw),]

#compute correlation between NES of PID pathways --> need to transpose it first
#pearson
KIRC_top_pw_cor_pearson = cor(t(KIRC_top_pw), method = "pearson")
Heatmap(KIRC_top_pw_cor_pearson, name = "mat", cluster_rows = FALSE, cluster_columns = FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 10))

#spearman
KIRC_top_pw_cor_spearman= cor(t(KIRC_top_pw), method = "spearman")
Heatmap(KIRC_top_pw_cor_spearman, name = "mat", cluster_rows = FALSE, cluster_columns = FALSE, column_names_gp = gpar(fontsize = 10), row_names_gp = gpar(fontsize = 10))

#add new row according to cluster, if patient in cluster 1 --> 1 (immune infiltrated), if in cluster 2 --> 0
cluster_number = vector(mode = 'list',72)
KIRC_patients = as.data.frame(colnames(KIRC_top_pw))
for (i in 1:ncol(KIRC_top_pw)){
  if (KIRC_patients[i,] %in% PID_c1_names){
    cluster_number[[i]] = 1
  } else {
    cluster_number[[i]] = 0
  }
}
cluster_numbers = list.cbind(cluster_number)
colnames(cluster_numbers) = colnames(KIRC_top_pw)
KIRC_top_pw = rbind(KIRC_top_pw, cluster_numbers)
rownames(KIRC_top_pw)[rownames(KIRC_top_pw) == "1"] <- "Infiltration"
KIRC_top_pw_t = as.data.frame(t(KIRC_top_pw))

#try simple logistic test regression

# Null-Modell
model0 = glm(Infiltration ~ 1, data = KIRC_top_pw_t)
summary(model0)

# Test-Modell
model1 = glm(Infiltration ~ PID_CD8_TCR_PATHWAY + PID_IL27_PATHWAY + PID_FOXM1_PATHWAY, data = KIRC_top_pw_t) #nur signifikante pw genommen (Beitrag leisten)
summary(model1)

#Omnibus-Test
modelchi = model1$null.deviance - model1$deviance
chidf = model1$df.null -model1$df.residual
chisqp = 1 - pchisq(modelchi, chidf) #0.0011 <0.05 test-Modell leistet signifikanten Erklärungsbeitrag

#Gütemaße
n = length(model1$residuals)
R2cs = 1-exp((model1$deviance - model1$null.deviance)/n)
R2n = R2cs/(1-exp(-(model1$null.deviance/n))) #0.91

#Plot
CD8_regression_plot = KIRC_top_pw_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_CD8_TCR_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_CD8_TCR_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

IL27_regression_plot = KIRC_top_pw_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_IL27_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_IL27_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

FOXM1_regression_plot = KIRC_top_pw_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_FOXM1_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_FOXM1_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

regression_plots = list(CD8_regression_plot, IL27_regression_plot, FOXM1_regression_plot)
ggarrange(plotlist = regression_plots, ncol = 3, nrow = 2)
```


```{r testing model1 (trained on small KIRC) on big KIRC TCGA dataset}
#try to test model1 on KIRC data from TCGA (531 samples)
KIRC_tcga = pathway_activity_matrix %>% dplyr::select(starts_with('KIRC'))
KIRC_tcga_PID = KIRC_tcga[PID_sigpw,]
KIRC_tcga_PID_t = data.frame(t(KIRC_tcga_PID))

#predict with model1
probabilities_2 <- model1 %>% predict(KIRC_tcga_PID_t, type = "response")
predicted.classes_2 <- data.frame(ifelse(probabilities_2 > 0.5, "1", "0"))
colnames(predicted.classes_2) = "prediction"
predicted.classes_2 = t(predicted.classes_2)
sum(as.numeric(predicted.classes_2[1,])) # 92 immune infiltrated patients

#create list with all patients who have a 1 in "prediction"
KIRC_tcga_immune_patients = numeric()
KIRC_tcga_PID_predict = rbind(KIRC_tcga_PID, predicted.classes_2)
for (i in 1:ncol(KIRC_tcga_PID_predict)){
  if (KIRC_tcga_PID_predict[11, i] == "1")
    KIRC_tcga_immune_patients = append(KIRC_tcga_immune_patients, colnames(KIRC_tcga_PID_predict[i]))
}

#remove first letters (KIRC.) from sample ID
KIRC_tcga_immune_patients_r = str_sub(KIRC_tcga_immune_patients, 6)

#create vector with all sample IDs that are in immune infiltrated cluster of PID clustering in big KIRC dataset
immuncluster_dendro = colnames(c1_patients)

#add new row according to clustering, if in immune cluster --> 1, else --> 0
cluster_number_tcga = vector(mode = 'list',531)
KIRC_tcga_patients = as.data.frame(str_sub(colnames(KIRC_tcga_PID_predict), 6))
for (i in 1:nrow(KIRC_tcga_patients)){
  if (KIRC_tcga_patients[i,] %in% immuncluster_dendro){
    cluster_number_tcga[[i]] = 1
  } else {
    cluster_number_tcga[[i]] = 0
  }
}
cluster_number_tcga = list.cbind(cluster_number_tcga)
colnames(cluster_number_tcga) = colnames(KIRC_tcga_PID_predict)
KIRC_tcga_PID_predict_cluster = rbind(KIRC_tcga_PID_predict, cluster_number_tcga)
rownames(KIRC_tcga_PID_predict_cluster)[rownames(KIRC_tcga_PID_predict_cluster) == "1"] <- "Infiltration"

#accuracy? 
mean(KIRC_tcga_PID_predict_cluster[11,] == KIRC_tcga_PID_predict_cluster[12,]) #90,58%!

#get annotations
KIRC_tcga_immunep_annotations = tcga_tumor_annotation %>% filter(sample %in% KIRC_tcga_immune_patients)

#check with immune deconvolution again
KIRC_tcga_tpm = 2^tcga_exp_KIRC
ensemblids_tcga_cleaned = rownames(KIRC_tcga_tpm)
symbols_tcga_cleaned = mapIds(org.Hs.eg.db, keys = ensemblids_tcga_cleaned, keytype = "ENSEMBL", column="SYMBOL") #336 duplicated
geneIDs <- ensembldb::select(EnsDb.Hsapiens.v79, keys= ensemblids_tcga_cleaned, keytype = "GENEID", columns = c("SYMBOL","GENEID")) #only 9775 gene symbols found --> remove missing genes from data frame

#remove duplicated gene symbols (24)
geneIDs_unique = distinct(geneIDs,SYMBOL, .keep_all= TRUE)
ensemblid_found = as.vector(geneIDs_unique$GENEID)
KIRC_tcga_tpm_filtered = KIRC_tcga_tpm %>% dplyr::filter(rownames(KIRC_tcga_tpm) %in% ensemblid_found)
rownames(KIRC_tcga_tpm_filtered) = geneIDs_unique$SYMBOL
KIRC_tcga_tpm_matrix = as.matrix(KIRC_tcga_tpm_filtered)

#immune deconvolution
quantiseq = deconvolute(KIRC_tcga_tpm_matrix, "quantiseq", tumor = "true")
cell_types_2 = quantiseq$cell_type
quantiseq_matrix = quantiseq[,-1]
rownames(quantiseq_matrix) = cell_types_2
quantiseq_matrix = as.matrix(quantiseq_matrix)

#remove unknown cell types, so that heatmap shows better colour scaling
quantiseq_matrix = quantiseq_matrix[-c(11),]

#plot immune deconvolution small KIRC
Heatmap_quantiseq_KIRC_tcga = Heatmap(quantiseq_matrix, name = "mat", column_km = 3, cluster_rows = FALSE, column_names_gp = gpar(fontsize = 2), row_names_gp = gpar(fontsize = 10),width = ncol(quantiseq_matrix)*unit(0.45, "mm"), height = nrow(quantiseq_matrix)*unit(10, "mm"), column_dend_height = unit(15, "mm"), column_title = "Immune cell fractions in KIRC TCGA")
Heatmap_quantiseq_KIRC_tcga
```

```{r logistic regression models trained on big KIRC dataset}
#second logistic regression model based on KIRC TCGA patients
#KIRC TCGA dataframe with only significant pathways (between clusters)
PID_sigpw_tcga = c("PID_CD8_TCR_PATHWAY", "PID_TCR_PATHWAY", "PID_IL1_PATHWAY", "PID_CD8_TCR_DOWNSTREAM_PATHWAY", "PID_IL27_PATHWAY", "PID_BCR_5PATHWAY", "PID_IL2_1PATHWAY", "PID_CXCR4_PATHWAY","PID_IL2_PI3K_PATHWAY","PID_IL2_STAT5_PATHWAY")

#extract pathway activity for KIRC TCGA patients for these 10 pathways
KIRC_tcga_PID_sigpw = KIRC_tcga[PID_sigpw_tcga,]

#add new row according to cluster, if in cluster 1 --> 1, if in cluster 2 --> 0
KIRC_tcga_PID_sigpw = rbind(KIRC_tcga_PID_sigpw, cluster_number_tcga)
rownames(KIRC_tcga_PID_sigpw)[rownames(KIRC_tcga_PID_sigpw) == "1"] <- "Infiltration"

#transpose dataframe for regression
KIRC_tcga_PID_sigpw_t = as.data.frame(t(KIRC_tcga_PID_sigpw))

#try simple logistic test regression

# Null-Modell
model0_tcga = glm(Infiltration ~ 1, data = KIRC_tcga_PID_sigpw_t)
summary(model0_tcga)

#  1. Test-Modell
model1_tcga = glm(Infiltration ~ PID_TCR_PATHWAY + PID_IL1_PATHWAY + PID_IL27_PATHWAY, data = KIRC_tcga_PID_sigpw_t) 
#meist signifikante pw genommen, wenn man alle einsetzen würde
summary(model1_tcga)
###Omnibus-Test
modelchi_tcga = model1_tcga$null.deviance - model1_tcga$deviance
chidf_tcga = model1_tcga$df.null -model1_tcga$df.residual
chisqp_tcga = 1 - pchisq(modelchi_tcga, chidf_tcga) # 2.829e-08 test-Model leistet signifikanten Erklärungsbeitrag

# 2. Test-Model
model2_tcga = glm(Infiltration ~ PID_CD8_TCR_PATHWAY + PID_IL1_PATHWAY, data = KIRC_tcga_PID_sigpw_t) 
#top 2 unkorrelierte pathways
summary(model2_tcga)
###Omnibus-Test
modelchi_tcga_2 = model2_tcga$null.deviance - model2_tcga$deviance
chidf_tcga_2 = model2_tcga$df.null -model2_tcga$df.residual
chisqp_tcga_2 = 1 - pchisq(modelchi_tcga_2, chidf_tcga_2) # 1.519e-08 test-Model leistet signifikanten Erklärungsbeitrag --> besser als erstes


# 3. Test-Model, nur PID_TCR_PATHWAY und PID_IL1_PATHWAY
model3_tcga = glm(Infiltration ~ PID_TCR_PATHWAY+ PID_IL1_PATHWAY, data = KIRC_tcga_PID_sigpw_t) 
summary(model3_tcga)
###Omnibus-Test
modelchi_tcga_3 = model3_tcga$null.deviance - model3_tcga$deviance
chidf_tcga_3 = model3_tcga$df.null -model3_tcga$df.residual
chisqp_tcga_3 = 1 - pchisq(modelchi_tcga_3, chidf_tcga_3) # 1.195e-08 test-Model leistet signifikanten Erklärungsbeitrag --> besser als erstes und zweites
```

```{r test logictic regression model(3) on small KIRC dataset}
#predict infiltration with second logistic regression model(3) from TCGA on small KIRC data set
#get KIRC pathway activity matrix for PID pathways
KIRC_for_model = as.data.frame(t(KIRC_PID_cl))

probabilities <- model3_tcga %>% predict(KIRC_for_model, type = "response")
predicted.classes <- data.frame(as.numeric(ifelse(probabilities > 0.5, "1", "0")))
colnames(predicted.classes) = "Prediction"
rownames(predicted.classes) = colnames(KIRC_PID_cl)
sum(as.numeric(predicted.classes[,1])) #41 predicted patients
KIRC_predict = rbind(KIRC_PID_cl, t(predicted.classes), cluster_numbers)
rownames(KIRC_predict)[rownames(KIRC_predict) == "1"] <- "Infiltration"

#accuracy? 
mean(KIRC_predict["Prediction",] == KIRC_predict["Infiltration",]) #0.9305556
KIRC_predict_t = as.data.frame(t(KIRC_predict))

#Plot
TCR_regression_plot = KIRC_predict_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_TCR_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_TCR_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

IL1_regression_plot = KIRC_predict_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_IL1_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_IL1_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

regression_plots_2 = list(TCR_regression_plot, IL1_regression_plot)
ggarrange(plotlist = regression_plots_2, ncol = 2, nrow = 2)
```

```{r Check performance of model3 with confusion table and ROC-curve}
#Confusion table and FDR-plots

#create dataframe with Infiltration and prediction only
predict_vs_infiltr = KIRC_predict_t[,89:90]
PI_table = table(predict_vs_infiltr)
PI_tibble = as_tibble(PI_table)
#confusion matrix
plot_confusion_matrix(PI_tibble, target_col = "Infiltration", 
                      prediction_col = "Prediction",
                      counts_col = "n")

#ROC-curve
pred = prediction(probabilities, KIRC_predict_t$Infiltration)
roc = performance(pred,"tpr","fpr")
plot(roc, lwd = 2)
abline(a = 0, b = 1) 
#calculate area under the curve
auc = performance(pred, measure = "auc")
print(auc@y.values) #0.9789062
```


```{r tried logistic regression model with gene expression values}
#try with gene expression?
PID_CD8_TCR_DOWNSTREAM_GENES = metabolic_genesets_PID_cl[["PID_CD8_TCR_DOWNSTREAM_PATHWAY"]]
KIRC_tcga_exp_PIDgenes = tcga_exp_KIRC[PID_CD8_TCR_DOWNSTREAM_GENES,] #81 genes
KIRC_tcga_exp_PIDgenes = na.omit(KIRC_tcga_exp_PIDgenes) #32 genes
cluster_numbers_tcga = cluster_numbers
colnames(cluster_numbers_tcga) = colnames(KIRC_tcga_exp_PIDgenes)
KIRC_tcga_exp_PIDgenes = rbind(KIRC_tcga_exp_PIDgenes, cluster_numbers_tcga)
rownames(KIRC_tcga_exp_PIDgenes)[rownames(KIRC_tcga_exp_PIDgenes) == "1"] <- "Infiltration"

KIRC_tcga_exp_PIDgenes_t = as.data.frame(t(KIRC_tcga_exp_PIDgenes))

#try simple logistic test regression

# Null-Modell
model0_tcga_genes = glm(Infiltration ~ 1, data = KIRC_tcga_exp_PIDgenes_t)
summary(model0_tcga_genes)

# Test-Modell
model1_tcga_genes = glm(Infiltration ~  ENSG00000138378 + ENSG00000049249, data = KIRC_tcga_exp_PIDgenes_t) #nur signifikante pw genommen (Beitrag leisten)
summary(model1_tcga_genes)

###Omnibus-Test
modelchi_tcga_genes = model1_tcga_genes$null.deviance - model1_tcga_genes$deviance
chidf_tcga_genes = model1_tcga_genes$df.null -model1_tcga_genes$df.residual
chisqp_tcga_genes = 1 - pchisq(modelchi_tcga_genes, chidf_tcga_genes) # 0.53 test-Modell leistet keinen signifikanten Erklärungsbeitrag

#sig : (ENSG00000172116, mit dem nicht signifikant), ENSG00000138378 (STAT4 =  increases the Th1 cells differentiation, cytotoxicity and IFNγ production of immune cells), ENSG00000049249(TNF receptor superfamily member 9, for CD8+ T cell activation)
```


