---
title: "Logistic regression"
author: "Maja Glotz"
date: "6/28/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(69)
```

```{r load packages}
library(immunedeconv)
library("org.Hs.eg.db")
library(gplots)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(rlist)
library(stats)
library(tidyselect)
library(cvms)
library(ROCR)
library(stringr)
library(car)
```

```{r loading data}
pathway_activity_matrix <- readRDS("data/pathway_activity_matrix.rds")
tcga_KIRC_tumor_cl <- readRDS("data/tcga_KIRC_tumor_cl.rds")
PID_c1_names <- readRDS("data/PID_c1_names.rds")
PID_c2_names <- readRDS("data/PID_c2_names.rds")
c1_patients <- readRDS("/data/c1_patients.RDS")
```

```{r adding cluster number and immunedeconv KIRC TCGA}
#KIRC data from TCGA (531 samples)
KIRC_tcga = pathway_activity_matrix %>% dplyr::select(starts_with('KIRC'))

#create vector with all sample IDs that are in immune infiltrated cluster of PID clustering in big KIRC dataset
immuncluster_dendro = colnames(c1_patients)

#add new row according to clustering, if in immune cluster --> 1, else --> 0
cluster_number_tcga = vector(mode = 'list',531)
KIRC_tcga_patients = as.data.frame(str_sub(colnames(KIRC_tcga), 6))
for (i in 1:nrow(KIRC_tcga_patients)){
  if (KIRC_tcga_patients[i,] %in% immuncluster_dendro){
    cluster_number_tcga[[i]] = 1
  } else {
    cluster_number_tcga[[i]] = 0
  }
}
cluster_number_tcga = list.cbind(cluster_number_tcga)
colnames(cluster_number_tcga) = colnames(KIRC_tcga_PID)

#get annotations
KIRC_tcga_immunep_annotations = tcga_tumor_annotation %>% filter(sample %in% KIRC_tcga_immune_patients)
```

```{r logistic regression model trained on big KIRC dataset}
#logistic regression model based on KIRC TCGA patients

#KIRC TCGA dataframe with only significant pathways (between clusters)
PID_sigpw_tcga = c("PID_CD8_TCR_PATHWAY", "PID_TCR_PATHWAY", "PID_IL1_PATHWAY", "PID_CD8_TCR_DOWNSTREAM_PATHWAY", "PID_IL27_PATHWAY", "PID_BCR_5PATHWAY", "PID_IL2_1PATHWAY", "PID_CXCR4_PATHWAY","PID_IL2_PI3K_PATHWAY","PID_IL2_STAT5_PATHWAY")

#extract pathway activity for KIRC TCGA patients for these 10 pathways
KIRC_tcga_PID_sigpw = KIRC_tcga[PID_sigpw_tcga,]

#add new row according to cluster, if in cluster 1 --> 1, if in cluster 2 --> 0
KIRC_tcga_PID_sigpw = rbind(KIRC_tcga_PID_sigpw, cluster_number_tcga)
rownames(KIRC_tcga_PID_sigpw)[rownames(KIRC_tcga_PID_sigpw) == "1"] <- "Infiltration"

#transpose dataframe for regression
KIRC_tcga_PID_sigpw_t = as.data.frame(t(KIRC_tcga_PID_sigpw))

#try simple logistic test regression

# Null-Modell
model0 = glm(Infiltration ~ 1, data = KIRC_tcga_PID_sigpw_t, family = binomial())
summary(model0)

#Test-Model, nur PID_TCR_PATHWAY und PID_IL1_PATHWAY
model = glm(Infiltration ~ PID_TCR_PATHWAY+ PID_IL1_PATHWAY, data = KIRC_tcga_PID_sigpw_t, family = binomial()) 
summary(model)

#Omnibus-Test
modelchi = model$null.deviance - model$deviance
chidf = model$df.null - model$df.residual
chisqp = 1 - pchisq(modelchi, chidf) # 0?

#Gütemaße
n = length(model$residuals)
R2nk = 1-(model$deviance / model$null.deviance) # McFadden 0.507
R2cs = 1-exp((model$deviance - model$null.deviance)/n) # cox snell 0.401
R2n = R2cs/(1-exp(-(model$null.deviance/n))) #Nagelkerke 0.63

#check with package
PseudoR2(model4_tcga, which = "McFadden") #0.5074508 (0.2-0.4 excellent fit?)
PseudoR2(model4_tcga, which = "CoxSnell") #0.4010591
PseudoR2(model4_tcga, which = "Nagelkerke") #0.6307655

#Odds-Ratios df
OR = exp(cbind(OR = coef(model), confint(model)))

#odds-ratios
exp(coefficients(model)[2]) #TCR: 2.64517
exp(coefficients(model)[3]) #IL-1: 3.206149

#if NES of PID_TCR rises 1 unit, the chance of being immune infiltrated, rises by 2.65
#if NES of PID_IL1 rises 1 unit, the chance of being immune infiltrated, rises by 3.2

#Wald-test for significance of each coefficient
wald = Anova(model, type="II", test="Wald")

#                 Df  Chisq Pr(>Chisq)    
#PID_TCR_PATHWAY  1 23.036  1.590e-06 ***
#PID_IL1_PATHWAY  1 36.289  1.701e-09 *

#Chisq-test for whole model
chisq = anova(model0,
              model,    # p-value: 7.860549e-60
      test="Chisq")

#Model 1: Infiltration ~ PID_TCR_PATHWAY + PID_IL1_PATHWAY
#Model 2: Infiltration ~ 1
#  Resid. Df Resid. Dev Df Deviance  Pr(>Chi)    
#1       528     264.19                          
#2       530     536.38 -2  -272.19 7.860549e-60***
```

```{r Regression plots model DF1}
#Plot logistic functions
TCR_regression_plot1 = KIRC_tcga_PID_sigpw_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_TCR_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_TCR_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

IL1_regression_plot1 = KIRC_tcga_PID_sigpw_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_IL1_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_IL1_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

regression_plots1 = list(TCR_regression_plot1, IL1_regression_plot1)
ggarrange(plotlist = regression_plots1, ncol = 2, nrow = 2)
IL1_regression_plot1
```

```{r test logictic regression model on small KIRC dataset}
#predict infiltration with second logistic regression model from TCGA on small KIRC data set
#get KIRC pathway activity matrix for PID pathways
KIRC_for_model = as.data.frame(t(KIRC_PID_cl))

probabilities <- model %>% predict(KIRC_for_model, type = "response")
predicted.classes <- data.frame(as.numeric(ifelse(probabilities > 0.5, "1", "0")))
colnames(predicted.classes) = "Prediction"
rownames(predicted.classes) = colnames(KIRC_PID_cl)
sum(as.numeric(predicted.classes[,1])) #42 predicted patients

#create vector with cluster numbers of PID clustering in small KIRC DS2
#if patient in cluster 1 --> 1 (immune infiltrated), if in cluster 2 --> 0
cluster_number = vector(mode = 'list',72)
KIRC_patients = as.data.frame(colnames(KIRC_PID_cl))
for (i in 1:ncol(KIRC_PID_cl)){
  if (KIRC_patients[i,] %in% PID_c1_names){
    cluster_number[[i]] = 1
  } else {
    cluster_number[[i]] = 0
  }
}
cluster_numbers = list.cbind(cluster_number)
colnames(cluster_numbers) = colnames(KIRC_PID_cl)

#add two new rows, prediction & cluster number
KIRC_predict = rbind(KIRC_PID_cl, t(predicted.classes), cluster_numbers)
rownames(KIRC_predict)[rownames(KIRC_predict) == "1"] <- "Infiltration"

#accuracy? 
mean(KIRC_predict["Prediction",] == KIRC_predict["Infiltration",]) #0.9444444
KIRC_predict_t = as.data.frame(t(KIRC_predict))
```

```{r Regression plots DF2}
#Plot logistic function
TCR_regression_plot = KIRC_predict_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_TCR_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_TCR_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

IL1_regression_plot = KIRC_predict_t %>%
        mutate(prob = ifelse(Infiltration == 1, 1, 0)) %>%
        ggplot(aes(PID_IL1_PATHWAY, prob)) +
        geom_point(alpha = 0.2) +
        geom_smooth(method = "glm", method.args = list(family = "binomial")) +
        labs(
          title = "Logistic Regression Model", 
          x = "PID_IL1_PATHWAY activity",
          y = "Probability of immune infiltration"
          )

regression_plots = list(TCR_regression_plot, IL1_regression_plot)
ggarrange(plotlist = regression_plots, ncol = 2, nrow = 2)
```

```{r Confusion table and ROC-curve}
#Confusion table and FDR-plots for model

#create dataframe with Infiltration and prediction only
predict_vs_infiltr = KIRC_predict_t[,89:90]
PI_table = table(predict_vs_infiltr)
PI_tibble = as_tibble(PI_table)

#confusion matrix
plot_confusion_matrix(PI_tibble, target_col = "Infiltration", 
                      prediction_col = "Prediction",
                      counts_col = "n")

#ROC-curve
pred = prediction(probabilities, KIRC_predict_t$Infiltration)
roc = performance(pred,"tpr","fpr")
plot(roc, lwd = 2)
abline(a = 0, b = 1) 

#calculate area under the curve
auc = performance(pred, measure = "auc")
print(auc@y.values) #0.9789062
```

```{r tried logistic regression model with gene expression values}

#try with gene expression?
PID_CD8_TCR_DOWNSTREAM_GENES = metabolic_genesets_PID_cl[["PID_CD8_TCR_DOWNSTREAM_PATHWAY"]]
KIRC_tcga_exp_PIDgenes = tcga_exp_KIRC[PID_CD8_TCR_DOWNSTREAM_GENES,] #81 genes
KIRC_tcga_exp_PIDgenes = na.omit(KIRC_tcga_exp_PIDgenes) #32 genes
cluster_numbers_tcga = cluster_numbers
colnames(cluster_numbers_tcga) = colnames(KIRC_tcga_exp_PIDgenes)
KIRC_tcga_exp_PIDgenes = rbind(KIRC_tcga_exp_PIDgenes, cluster_numbers_tcga)
rownames(KIRC_tcga_exp_PIDgenes)[rownames(KIRC_tcga_exp_PIDgenes) == "1"] <- "Infiltration"

KIRC_tcga_exp_PIDgenes_t = as.data.frame(t(KIRC_tcga_exp_PIDgenes))

#try simple logistic test regression

# Null-Modell
model0_tcga_genes = glm(Infiltration ~ 1, data = KIRC_tcga_exp_PIDgenes_t)
summary(model0_tcga_genes)

# Test-Modell
model1_tcga_genes = glm(Infiltration ~  ENSG00000138378 + ENSG00000049249, data = KIRC_tcga_exp_PIDgenes_t) #nur signifikante pw genommen (Beitrag leisten)
summary(model1_tcga_genes)

###Omnibus-Test
modelchi_tcga_genes = model1_tcga_genes$null.deviance - model1_tcga_genes$deviance
chidf_tcga_genes = model1_tcga_genes$df.null -model1_tcga_genes$df.residual
chisqp_tcga_genes = 1 - pchisq(modelchi_tcga_genes, chidf_tcga_genes) # 0.53 test-Modell leistet keinen signifikanten Erklärungsbeitrag

#sig : (ENSG00000172116, mit dem nicht signifikant), ENSG00000138378 (STAT4 =  increases the Th1 cells differentiation, cytotoxicity and IFNγ production of immune cells), ENSG00000049249(TNF receptor superfamily member 9, for CD8+ T cell activation)
```


